---
title: "Elsa-analysis1"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# 1. Set up working Environment

## 1.1 Load required libraries:
```{r libraries, echo=FALSE, message=FALSE, warning=FALSE}
## Load required libraries
library(ggnewscale)
library(ggtree)
library(R.utils)
library(tidyverse)
library(ape)
library(devtools)
library(ggplot2)
library(hillR)
library(spaa)
library(vegan)
library(hilldiv2) # note use of hilldiv2 rather than hilldiv
library(Rhdf5lib)
library(phyloseq)
library(phytools)
library(microbiome) 
library(matrixStats)
library(microbiomeutilities) 
library(lme4)
library(MuMIn)
library(nlme)
library(knitr) 
library(kableExtra)
library(pairwiseAdonis)
library(sjPlot)
library(distillR) 
library(RColorBrewer)
library(reshape2)
library(ggpubr)
library(ggdendro)
library(grid)
library(gplots)
library(dendextend)
library(stringr) 
library(Rtsne) 
library(glue) 
library(ggtree)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(ggtreeExtra)
library(DESeq2)
library(ANCOMBC)
library(lefser)
library(dplyr)
library(mia)
library(pheatmap)
```
##1.2 Establish workflow variables directories
```{r directories, comment="", echo=FALSE, message=FALSE, warning=FALSE}
## set variable names for files, and working directory

workingdir="/Users/elsa/Desktop/THESIS/sled_dog_parasite"
counts_file="/Users/elsa/Desktop/THESIS/sled_dog_parasite/DMB0032_counts.tsv"
tree_file="/Users/elsa/Desktop/THESIS/sled_dog_parasite/DMB0032.tree"
taxonomy_file="/Users/elsa/Desktop/THESIS/sled_dog_parasite/DMB0032_mag_info.tsv"
metadata_file="/Users/elsa/Desktop/THESIS/sled_dog_parasite/DMB0032_metadata.tsv"
coverage_file="/Users/elsa/Desktop/THESIS/sled_dog_parasite/DMB0032_coverage.tsv"
```
```{r loaddata, comment="", echo=FALSE, message=FALSE, warning=FALSE}
## Load data 
setwd(workingdir)

batch="DMB0032" # this specifies the batch of data within the EHI data framework

counts_table <- read.table(counts_file, sep="\t",row.names=1,header=T)
coverage_table <- read.table(coverage_file, sep="\t",row.names=1,header=T)
metadata <- read.table(metadata_file,sep="\t",header=T)%>%
	dplyr::rename(sample=EHI_plaintext)
mags_table <- read.table(taxonomy_file,sep="\t",header=T)
tree <- read.tree(tree_file)

# Load EHI taxonomy colours. Un-comment this line if download is necessary
#colours_URL="https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv"
#download.file(colours_URL, "ehi_phylum_colors.tsv")

ehi_phylum_colors <- read.table("~/Desktop/Thesis/R-analysis1/ehi_phylum_colors.tsv",sep="\t",header=T,comment.char = "")

#Fromat data for analysis, delete *__ from the taxonomy names
ehi_phylum_colors1 <- ehi_phylum_colors %>%
  mutate_at(vars(phylum), ~ stringr::str_replace(., "[dpcofgs]__", ""))
taxonomyclean <- mags_table%>%
  mutate_at(vars(domain,phylum,class,order,family,genus,species), ~ str_replace(., "[dpcofgs]__", ""))
```
```{r summary, echo=FALSE, message=FALSE, warning=FALSE}
nsamples <- ncol(counts_table) # define number of samples, by reading from the counts table
metagenomic_bases <- sum(metadata$metagenomic_bases)
host_bases <- sum(metadata$host_bases)
discarded_bases <- sum(round(((metadata$metagenomic_bases+metadata$host_bases)/(1-metadata$bases_lost_fastp_percent))-(metadata$metagenomic_bases+metadata$host_bases))) #define amount of discarded bases from quality filtering
total_bases <- discarded_bases + host_bases + metagenomic_bases
singlem_bases <- sum(metadata$metagenomic_bases * metadata$singlem_fraction) 
nmags <- nrow(counts_table) # establish number of MAGs for analysis
new_species <- mags_table %>%
	filter(species == "s__") %>%
	nrow()

sequencing_depth <- colSums(counts_table)
sequencing_depth_sum <- sum(sequencing_depth)
sequencing_depth_mean <- mean(sequencing_depth)
sequencing_depth_sd <- sd(sequencing_depth)
```

# 2 Data pre-processing

## 2.1 General statistics

**Number of samples in total**
```{r nsamples, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ncol(counts_table) # 58 samples
```
**Number of MAGs**
The number of metagenome-assembled genomes (MAG) or draft bacterial genomes reconstructed from the metagenomic data.

```{r nmags, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cat(nmags) # 555 MAGs
```
**Amount of total data (GB):**
The amount of total DNA data sequenced in gigabases (GB, one billion nucleotide bases).

```{r totalGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
totalgb <- round(total_bases / 1000000000,2) # convert to GB 
cat(totalgb) # 333.71 GB produced
```
**Amount of discarded data (GB):**
The amount of data discarded due to low quality or lack of informativeness during data preprocesing. Discarding 5-15% of the produced data is within the expected range, due to formation of adaptor dimers, inclusion of adaptors in sequencing reads due to short insert sizes, low sequencing quality, etc.

```{r discardedGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
discardgb <- round(discarded_bases / 1000000000,2) # quality filtering
cat(discardgb) # 10.37 GB discarded
```
**Amount of discarded data (in % of the raw data):**

```{r %discarded, comment="", echo=FALSE, message=FALSE, warning=FALSE}
discarddata <- round(discarded_bases / total_bases * 100,2) 
cat(discarddata) # 3.11% removed after quality filtering
```
**Amount of host data (GB):**
The amount of data mapped against the host genome. The percentage refers to the amount of data mapped to the host genome respect to quality-filtered data. Note that this value can be very variable depending on the biological features of the sample (e.g., anal swabs contain more host DNA than faeces) and the employed reference genome (e.g., the chances for mapping to the genome are lower as the distance between) the study species and the employed reference genome differ).

```{r hostGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
hostGB <- round(host_bases / 1000000000,2)
cat(hostGB) # 6.49 GB from host
```
**Amount of host data (% of the quality-filtered data):**

```{r host%, comment="", echo=FALSE, message=FALSE, warning=FALSE}
hostdata <- round(host_bases / (total_bases-discarded_bases) * 100,2)
cat(hostdata) # 2.01% of quality filtered data is from host
```
**Estimated prokaryotic data:** 
The amount and proportion of data belonging to prokayotic genomes respect to the total metagenomic fraction, as estimated from singleM analysis. Note that this is an estimation that relies on the genome sizes of genomes available in reference databases. If a given taxon is not properly represented, genome size estimations can be less accurate.

```{r prokaGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
prokaGB <- round(singlem_bases / 1000000000,2)
cat(prokaGB) # 293.14 GB prokaryotic data
```
**Estimated prokaryotic data (% of the metagenomic data):** 

```{r proka%, comment="", echo=FALSE, message=FALSE, warning=FALSE}
prokadata <- round(singlem_bases / (metagenomic_bases) * 100,2)
cat(prokadata) # 92.51% of metagenomic data is prokaryotic
```
**Amount of metagenomic data (GB):**
The amount of data mapped against the host genome. The percentage refers to the amount of data mapped to the host genome respect to quality-filtered data. Note that this value can be very variable depending on the biological features of the sample (e.g., anal swabs contain more host DNA than faeces) and the employed reference genome (e.g., the chances for mapping to the genome are lower as the distance between) the study species and the employed reference genome differ).

```{r metaGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
metaGB <- round(metagenomic_bases / 1000000000,2)
cat(metaGB) # 316.85 GB metagenomic data
```
**Amount of metagenomic data (% of the quality-filtered data):**

```{r meta%, comment="", echo=FALSE, message=FALSE, warning=FALSE}
metaperce <- round(metagenomic_bases / (total_bases-discarded_bases) * 100,2)
cat(metaperce) # 97.99%  quality filtered data is metagenomic
```
**Total mapped sequencing depth (million reads):**
The amount of reads (and nucleotide bases) that were mapped to the entire MAG catalogue. Note that the amount of bases is only an approximation estimated by multiplying the exact number of mapped reads by 250 bp.

```{r totalreads, comment="", echo=FALSE, message=FALSE, warning=FALSE}
totalreads <- round(sequencing_depth_sum / 1000000,2)
cat(totalreads) # 1860.26 million reads
```
**Total mapped sequencing depth (GB):**

```{r mappedGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
mappedGB <- round(sequencing_depth_sum / 1000000000 * 143,2)
cat(mappedGB) # 266.02 GB total sequencing depth
```
**Average mapped sequencing depth (million reads):** 
This is the average number of reads (and nucleotide bases) mapped to each sample. Note that the amount of bases is only an approximation estimated by multiplying the exact number of mapped reads by 250 bp.
```{r meanreads, comment="", echo=FALSE, message=FALSE, warning=FALSE}
meanreads <- round(sequencing_depth_mean / 1000000,2)
cat(meanreads) # 32.07 million reads avg seq depth
```
**Average mapped sequencing depth (GB):** 
```{r meanGB, comment="", echo=FALSE, message=FALSE, warning=FALSE}
meanGB <- round(sequencing_depth_mean / 1000000000 * 143,2)
cat(meanGB) # 4.59 GB avg seq depth
```

## 2.2 MAG catalogue

### 2.2.1 Phylogenetic tree
The phylogenetic tree is constructed by placing the MAG sequences within the reference archaeal and bacterial trees using GTDBTK, followed by merging both trees.

```{r list_phyla, echo=FALSE, warning=FALSE}
phyla <- ehi_phylum_colors1 %>%
  dplyr::right_join(taxonomyclean, by=join_by(phylum == phylum)) %>% 
	arrange(match(genome, tree$tip.label)) %>% 
  dplyr::select(phylum, colors) %>%
	unique()
```
```{r circular_tree_prep, echo=FALSE, warning=FALSE, comments="", message=FALSE, results="hide"}
heatmap <- ehi_phylum_colors1 %>%
  dplyr::right_join(taxonomyclean, by=join_by(phylum == phylum)) %>%
	arrange(match(genome, tree$tip.label)) %>%
  select(genome,phylum) %>%
	mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
	column_to_rownames(var = "genome")

colors_alphabetic <- ehi_phylum_colors1 %>%
  dplyr::right_join(taxonomyclean, by=join_by(phylum == phylum)) %>%
	arrange(match(genome, tree$tip.label)) %>%
  select(phylum, colors) %>%
	unique() %>%
	arrange(phylum) %>%
	select(colors) %>%
	pull()

circular_tree <- force.ultrametric(tree,method="extend") %>%
	ggtree(., layout = 'circular', size = 0.3)

circular_tree <- gheatmap(circular_tree, heatmap, offset=0.65, width=0.1, colnames=FALSE) +
		scale_fill_manual(values=colors_alphabetic) +
		geom_tiplab2(size=1, hjust=-0.1) +
		theme(plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

print(circular_tree)
```
### 2.2.2 MAG details
Overview of the taxonomy and genome characteristics of the MAGs.\
**Completeness:** completeness of the MAG according to CheckM assessment.\
**Contamination:** contamination or redundancy of the MAG according to CheckM assessment.\
**Size:** size of the MAG in megabases (MB, one million nucleotide bases).

```{r complet_conta, comment="", echo=FALSE, message=FALSE, warning=FALSE}
comp_cont <- mutate(
  select(taxonomyclean, genome, phylum, completeness, contamination, mag_size),
  mag_size = round(mag_size / 1000000, 2),
  comp = completeness,
  cont = contamination,
  size = mag_size
) %>%
  select(-completeness, -contamination, -mag_size) %>%
  remove_rownames() %>%
  arrange(match(genome, rev(tree$tip.label)))

```
```{r complet_conta_table, comment="", echo=FALSE, message=FALSE, warning=FALSE}
comp_cont_table <- comp_cont %>% 
  select(-genome) %>% 
  group_by(phylum) %>% 
  summarise_at(.vars = names(.)[c(2,3,4)],.funs = c(mean="mean", sd="sd"))
comp_cont_table <- comp_cont_table[,c(1,2,5,3,6,4,7)]
knitr::kable(comp_cont_table, format = "html", full_width = F,col.names = c("Phylum", "Completeness Mean", "Completeness SD", "Contamination Mean", "Contamination SD", "Size Mean", "Size SD"), digits = 2) %>%
  kable_styling(latex_options="scale_down")
```

```{r plot_mag_stats, echo=FALSE, warning=FALSE}
ggscatter(comp_cont, x = "comp", y = "cont", color="phylum", 
          add = "reg.line", conf.int = TRUE,  add.params = list(color = "black", fill = "lightgray"),
          cor.coef = TRUE, cor.method = "kendall", size = "size",
          cor.coeff.args = list(method = "kendall", label.x = 80, label.sep = "\n"), 
          xlab = "Completeness", ylab = "Contamination", legend="right") +
				scale_color_manual(values=colors_alphabetic) +
  guides(col=guide_legend("Phylum"),
         size=guide_legend("MAG size"))
```

## 2.3 Sequencing depth assessment
When performing genome-resolved metagenomic analyses on host-associated microbial communities, the data usually contains a mixture of origins.

One fraction is low-quality data that is discarded in the bioinformatic preprocessing due to lack of informativeness. These data include low-quality bases, adaptors, low-complexity reads and alike, which do not contribute to the study. Another fraction belongs to the host genome against which the data are mapped. The host fraction can be very variable depending on the species and the sample type, and while it is not informative for metagenomic analyses, it can be used for genomic analyses. The rest is what we call the metagenomic fraction. Part of the metagenomic fraction is built into draft bacterial genomes or MAGs, against which metagenomic reads are mapped later on to quantify relative representation of genomes. The fraction that is not built into MAGs is what is also unmapped against the MAG catalogue. This last fraction includes DNA dietary items, viruses and other organisms, but can values_to include prokaryotic DNA of bacteria and archaea that were unable to be reconstructed.

In order to have representative results, the number of reads mapped to the MAG catalogue should be similar across samples. However, multiple reasons can create large imbalances, including uneven sequencing depth, different microbiome complexity across samples, different amount of host or non-microbial reads in the dataset, etc. The following plot shows the distribution of reads across samples.

```{r data_fraction, warning=FALSE, echo=FALSE, fig.height=5}
# Calculate sequence fractions
sequence_fractions <- counts_table %>%
  rownames_to_column("Genome") %>%
  pivot_longer(-Genome, names_to = "sample", values_to = "value") %>%
  group_by(sample) %>%
  summarise(mags = sum(value)) %>%
	dplyr::left_join(metadata, by = join_by(sample == sample))  %>%
	select(sample,mags,metagenomic_bases,host_bases,bases_lost_fastp_percent) %>%
	mutate(mags_bases = mags*146) %>%
	mutate(lowqual_bases = ((metagenomic_bases+host_bases)/(1-bases_lost_fastp_percent))-(metagenomic_bases+host_bases)) %>%
	mutate(unmapped_bases = metagenomic_bases - mags_bases) %>%
	select(sample,mags_bases,unmapped_bases,host_bases,lowqual_bases)

mags_bases_mean <- sequence_fractions %>%
	mutate(mags_bases = mags_bases / 1000000000) %>%
	select(mags_bases) %>%
	mean()

sequence_fractions_pivot <- sequence_fractions %>%
	pivot_longer(!sample, names_to = "fraction", values_to = "value") %>%
	mutate(value = value / 1000000000) %>%
	mutate(fraction = factor(fraction, levels = c("lowqual_bases","host_bases","unmapped_bases","mags_bases")))

sequence_fractions_meta <- merge(sequence_fractions_pivot, metadata, by="sample")
```
```{r data_fraction_barplot, echo=FALSE, warning=FALSE}
ggplot(sequence_fractions_meta, aes(x = sample, y = value, fill=fraction)) +
  geom_bar(position="stack", stat = "identity") +
  scale_fill_manual(name=NULL,
                    breaks=c("lowqual_bases","host_bases","unmapped_bases","mags_bases"),
                    labels=c("Low quality","Host","Unmapped", "MAGs"),
                    values=c("#CCCCCC","#178a94","#ee8080","#d03161")) +
#  geom_hline(yintercept = mags_bases_mean, linetype = "dashed", color = "black") +
  facet_grid(~region, scale="free", space="free") + # grouped the samples by region
  labs(x = "Samples", y = "Amount of data (GB)") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6),
        strip.text.x = element_text(size = 7, face = "bold"),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.background = element_rect(fill = NA, color = "black"),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
        legend.position="bottom",
        legend.title=element_blank())

```

## 2.4 Minimum genome-coverage filtering
Mapping of sequencing reads against the reference genome catalogue is not perfect, and in consequence, all MAGs tend to get a few reads assigned. Implementing a minimum genome coverage filter aims at minimising artificial inflation of diversity due to this artifact of genome-resolved metagenomic analysis. However, if the sequencing depth is low and uneven across samples, this filtering can also introduce more distorsion.

```{r coverage, echo=FALSE, warning=FALSE}
# Coverage filtering
min_coverage=0.3
count_table_cov <- coverage_table %>%
  mutate(across(everything(), ~ ifelse(. > min_coverage, 1, 0))) %>%
  map2_df(., counts_table, ~ .x * .y) %>%
  as.data.frame()
rownames(count_table_cov) <- rownames(coverage_table)
```

## 2.5 Genome-size normalisation
Bacterial genomes can vary between 1 and 8 MB, which make relative representation of each genome dependent on its size. To account for genome size biases, read-counts can be normalised by applying a normalisation factor that modifies the read numbers according to the size of each genome compared to the average genome size in the dataset.

```{r genome_norm, echo=FALSE, warning=FALSE}
#Transform by mean MAG-size
mags_table1 <- column_to_rownames(mags_table, "genome")
genome_read_sizes <- mags_table1[rownames(count_table_cov),] %>%
    dplyr::select(mag_size) %>%
    mutate(mag_size = mag_size / 143) %>% #143 nt is the average read-length after quality filtering in EHI data
    pull()
count_table_cov_size <- sweep(count_table_cov, 1, genome_read_sizes, "/") #USE THIS COUNT TABLE FOR ALL DA ANALYSIS

count_table_cov_size_rel <- count_table_cov_size %>%
  rownames_to_column("Genome") %>%
  mutate_at(vars(-Genome),~./sum(.)) #TSS normalisation
```
## 2.6 Count table
Once low-coverage genome counts have been filtered out, and the read counts have been normalised into genome counts, we can visualise the relative MAG abundances per sample. Note that the count scale is log-transformed.
```{r cov_size_heatmap, echo=FALSE, warning=FALSE, comments="", message=FALSE, results="hide"}
vertical_tree <- force.ultrametric(tree,method="extend") %>%
		ggtree(., size = 0.3)

#Add phylum colors
vertical_tree <- gheatmap(vertical_tree, heatmap, offset=0, width=0.1, colnames=FALSE) +
	scale_fill_manual(values=colors_alphabetic)

#Reset fill scale
vertical_tree <- vertical_tree + new_scale_fill()

#Add counts
vertical_tree <- gheatmap(vertical_tree, log10(count_table_cov_size), offset=0.04, width=3.5, colnames=TRUE, colnames_angle=90, font.size=2, colnames_position="top", colnames_offset_y = 9) +
	vexpand(.08) +
	coord_cartesian(clip = "off") +
	scale_fill_gradient(low = "white", high = "steelblue", na.value="white")
```
```{r plot_tree, echo=FALSE, warning=FALSE, comments="", message=FALSE}
#Plot tree
vertical_tree +
	theme(legend.position='bottom',legend.key.size = unit(0.3, 'cm')) + labs(fill='Counts')
```

# 3 Taxonomic composition
Note that TSS normalisation simply divides each count value for the total count for the sample, thus transforming the data to 0-1 scale.

```{r taxo_comp, echo=FALSE, warning=FALSE}
count_table_cov_size_pivot <- count_table_cov_size %>%
  rownames_to_column("Genome") %>%
  mutate_at(vars(-Genome),~./sum(.)) %>% # TSS nornalisation
  pivot_longer(-Genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  dplyr::left_join(., mags_table, by = join_by(Genome == genome)) %>% #append taxonomy
  mutate(phylum = fct_relevel(phylum, rev(ehi_phylum_colors$phylum))) #sort phyla by taxonomy

count_table_cov_size_pivot <- count_table_cov_size_pivot %>%
  mutate_at(vars(domain,phylum,class,order,family,genus,species), ~ stringr::str_replace(., "[dpcofgs]__", "")) #delete __ from the taxonomy

count_table_cov_size_pivot_meta <- metadata[c(1,4)] %>%
    merge(., count_table_cov_size_pivot, by="sample") #merge with metadata

# Retrieve taxonomy colors to use standardised EHI colors
phylum_colors <- ehi_phylum_colors1 %>%
  filter(phylum %in% unique(count_table_cov_size_pivot$phylum)) %>%
  select(colors) %>%
  pull() %>%
  rev()
phylum_colors <- c(phylum_colors,"#cccccc") #REMOVE! ONLY FOR ARCHAEANS
```

```{r plot_another, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Plot stacked barplot
ggplot(count_table_cov_size_pivot_meta, aes(x=sample,y=count,fill=phylum, group=phylum))+ 
    geom_bar(stat="identity")+ 
    scale_fill_manual(values=phylum_colors) +
    facet_grid(~region, scale="free", space="free") +
    labs(x = "Samples", y = "Relative abundance") +
    guides(fill = guide_legend(ncol = 4)) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6),
        strip.text.x = element_text(size = 6, face = "bold"),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.background = element_rect(fill = NA, color = "black"),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
        legend.position="bottom",
        legend.title=element_blank(),
        legend.text = element_text(size=6),
        legend.key.size = unit(0.3, 'cm'))
```


```{r phyloseq, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
## Making phyloseq object -- needed for later DA analysis

# OTU table
MAGotu <- otu_table(count_table_cov_size, taxa_are_rows = T)

# Tax table
taxonomyclean1 <- taxonomyclean[,-1]
taxmatrix <- as.matrix(taxonomyclean1[1:8])
rownames(taxmatrix) <- taxonomyclean$genome
taxtable <- tax_table(taxmatrix) 

#Metadata
metadata <- na.omit(metadata)
rownames(metadata) <- NULL
metadata.pre <- column_to_rownames(metadata, "sample")

#Sample table
sample_tab <- sample_data(metadata.pre)

# Tree
tree <- phytools::force.ultrametric(tree, method = "extend")
treephylo = phyloseq::phy_tree(tree)

## Merge into phyloseq object
physeq <- phyloseq(MAGotu, treephylo, sample_tab,taxtable)
```

### Phylum percentages

```{r phylum1, comment="", echo=FALSE}

physeq_phylum <- microbiome::aggregate_taxa(physeq, 'phylum')
physeq_phylum_rel <-  microbiome::transform(physeq_phylum, "compositional")
table.rel1 <- physeq_phylum_rel@otu_table*100
means.table.rel1 <- as.data.frame(rowMeans(table.rel1))
sd.table.rel1 <- as.data.frame(matrixStats::rowSds(table.rel1, useNames = TRUE))
summary.phylum1 <- merge(means.table.rel1, sd.table.rel1, by="row.names")
colnames(summary.phylum1) <- c("Phylum","Mean", "SD")
print(summary.phylum1[order(-summary.phylum1$Mean),], row.names = FALSE)
```

### genus percentages

```{r genus1, comment="", echo=FALSE}

physeq_genus <- microbiome::aggregate_taxa(physeq, 'genus')
physeq_genus_rel <-  microbiome::transform(physeq_genus, "compositional")
table.rel2 <- physeq_genus_rel@otu_table*100
means.table.rel2 <- as.data.frame(rowMeans(table.rel2))
sd.table.rel2 <- as.data.frame(matrixStats::rowSds(table.rel2, useNames = TRUE))
summary.genus1 <- merge(means.table.rel2, sd.table.rel2, by="row.names")
colnames(summary.genus1) <- c("genus","Mean", "SD")
print(summary.genus1[order(-summary.genus1$Mean),], row.names = FALSE)
```

# 4 Alpha diversity calculations

Diversity estimations for each sample.\
**Richness:** Number of MAGs per sample (after applying coverage filter).\
**Neutral diversity:** Hill number of q=1 (Shannon diversity), a diversity metric that accounts for richness and eveness (relative abundances) of the MAGs.\
**Phylogenetic diversity:** Phylogenetic Hill number of q=1, a diversity metric that accounts for richness and eveness (relative abundances), as well as phylogenetic relationships among MAGs.\
**Functional diversity:** Functional Hill number of q=1, a diversity metric that accounts for richness and eveness (relative abundances), as well as functional dissimilarities among MAGs.\

```{r additional meta, comment="", echo=TRUE, message=FALSE, warning=FALSE}
count_filtered <- column_to_rownames(count_table_cov_size_rel, "Genome")

metadata_unified <- read.csv("/Users/elsa/Desktop/THESIS/Thesis/metadata_unified.csv")
metadata_unified_clean <- metadata_unified %>%
  filter(sample != "")
selected_columns <- metadata_unified_clean %>%
  dplyr::select(sample, Q_code, PR_results_short, PR_toxocaris_leonina, PR_sarcocystis, PR_Tania_serialis, PR_cryptosporidium, PR_giardia, sero_toxoplasma_gondii_PR, sero_trichincella_spp_PR, Huldscore, Eukaryota_., Cryptosporidium_., Giardia_., Sarcocystis_., Toxoplasma_gondii_., Toxacaris_., Trichinella_., Uncinaria_., Echinococcus_., Taenia_., Cestoda_., Nematoda_., Virus_., PARVO_Parvoviridae_., PARVO_Bocaparvovirus_., PARVO_Canine_Minute_., PARVO_Canine_bufavirus_., PARVO_Canine_protoparvovirus_., PARVO_Canine_parvovirus_., ADENO_Adenoviridae_., ADENO_Canine_mastadenovirus_., PAP_Papillomaviridae_., )

all_metadata <- dplyr::left_join(metadata, selected_columns, by = "sample") # use for remaining analysis
```

### 4.1 Neutral
```{r alpha_div_neutral, comment="", echo=TRUE, message=FALSE, warning=FALSE}
alpha_div_neutral <- hilldiv(count_filtered, q=1)
```
#### Average neutral alpha diversities (q1)
```{r div_mean, comment="", echo=TRUE, message=FALSE, warning=FALSE}

alpha_div_N <- t(alpha_div_neutral) %>%
  as.data.frame() %>%
  tibble::rownames_to_column("sample") %>%
  merge(., all_metadata, by="sample")

table_mean_alpha <- alpha_div_N %>% 
  group_by(region) %>% 
  summarise_at(.vars = names(.)[2],.funs = c(mean="mean", sd="sd"))
knitr::kable(table_mean_alpha, format = "html", full_width = F,col.names = c('Groups', 'Mean', 'SD'), digits = 3) %>%
  kable_styling(latex_options="scale_down")
```

```{r alpha_table, comment="", echo=TRUE, message=FALSE, warning=FALSE}
alpha_div <- t(alpha_div_neutral) %>%
  as.data.frame() %>%
  dplyr::rename( "Diversity" = "q1")  %>%
  rownames_to_column("sample") %>%
  dplyr::left_join(sequence_fractions, by = join_by(sample == sample)) %>% #add sequencing depth information
  mutate(depth=round(mags_bases/1000000,3))
```
#### Correlations
```{r correlation2_plot2, comment="", echo=TRUE, message=FALSE, warning=FALSE}

ggscatter(alpha_div, x = "mags_bases", y = "unmapped_bases", color="depth", 
          add = "reg.line", conf.int = TRUE,  add.params = list(color = "gray", fill = "lightgray"),
          cor.coef = TRUE, cor.method = "kendall", size = "Diversity",
          cor.coeff.args = list(method = "kendall", label.x = 3e+9, label.sep = "\n"), 
          xlab = "MAG bases", ylab = "Unmapped reads", legend="right") +
  guides(col=guide_legend("Depth"),
         size=guide_legend("Alpha diversity")) +
  font("legend.title", face = "bold")
```

#### Alpha diversity plots - Community Level
```{r alpha_div_neutral_plot1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# Box plots for Neutral alpha div

alpha_div_N %>%
  ggplot(aes(x = region, y = q1, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Neutral Alpha diversity")
```

```{r alpha_div_neutral_stats, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# Shapiro-Wilk test for normality
alpha_div_N %>%
  filter(region %in% c("Daneborg", "Ittoqqortoormii")) %>%
  group_by(region) %>%
  summarise(shapiro_p = shapiro.test(q1)$p.value) -> shapiro_results
# Results: p = 0.01942486	(Daneborg), p = 0.61532713	Ittoqqortoormiit
# THEREFORE Daneborg data is not normally distributed, though Ittoq is. THUS a non-parametric test is applied

# Non-normally distributed data test for equal variance
car::leveneTest(q1 ~ region, alpha_div_N)
# Results: p = 0.0008284
# THEREFORE data does not exhibit equal variance. 

# Wilcoxon signed rank test for statistical significance
alpha_div_N %>%
  filter(region %in% c("Daneborg", "Ittoqqortoormii")) %>%
  group_by(region) %>%
  summarise(wilcox_p = wilcox.test(q1 ~ region, data = .)$p.value) -> wilcox_results
print(wilcox_results)
# Results: p = 0.4869263	
# THEREFORE Neutral alpha diversity does NOT have statistically significant differences between Ittoq and Daneborg
```

### 4.2 Phylogenetic level
```{r alpha_div_phylo, comment="", echo=TRUE, message=FALSE, warning=FALSE}
alpha_div_phylo <- hilldiv(data=count_filtered,tree=tree, q=1)
```

#### Average phylogenetic alpha diversities
```{r div_P_mean, comment="", echo=TRUE}

alpha_div_P <-  t(alpha_div_phylo) %>%
  as.data.frame() %>%
  tibble::rownames_to_column("sample") %>%
  merge(., all_metadata, by="sample")

table_mean_alpha_phylo <- alpha_div_P %>%
  group_by(region) %>% 
  summarise_at(.vars = names(.)[2],.funs = c(mean="mean", sd="sd"))

knitr::kable(table_mean_alpha_phylo, format = "html", full_width = F,col.names = c('Groups', 'Mean', 'SD'), digits = 3) %>%
  kable_styling(latex_options="scale_down")
```

#### Alpha diversity plots - Phylogenetic level
```{r alpha_div_phylo_plot1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# Box plot for phylogenetic alpha diversity 
alpha_div_P %>%
  ggplot(aes(x = region, y = q1, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Phylogenetic Alpha diversity")
```
```{r r alpha_div_phylo_stats, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# Shapiro-Wilk test for normality
alpha_div_P %>%
  filter(region %in% c("Daneborg", "Ittoqqortoormii")) %>%
  group_by(region) %>%
  summarise(shapiro_p = shapiro.test(q1)$p.value) -> shapiro_results_P
# Results: p = 0.4556244 (Daneborg), p = 0.4019311 Ittoqqortoormiit
# THEREFORE both regions exhibit NORMALLY distributed alpha_div_P values, and a PARAMETRIC test can be applied


# Using Bartlett to test for variance
bartlett.test(q1 ~ region, alpha_div_P)
# Results: p-value = 0.0004803
# THEREFORE therefore equal variance cannot be assumed


# T-test not assuming equal variance
alpha_div_P %>%
  filter(region %in% c("Daneborg", "Ittoqqortoormii")) %>%
  group_by(region) %>%
  summarise(t_test_p = t.test(q1 ~ region, data = ., var.equal = FALSE)$p.value) -> t_test_results_P
print(t_test_results_P)
# Results: p = 0.3423079	
# THEREFORE Phylogenetic alpha diversity is NOT have statistically significant differences between Ittoq and Daneborg
```

### Beta diversity - Community level
```{r beta_div_neutral, comment="", echo=TRUE, message=FALSE, warning=FALSE, results='hide'}
beta_colors <- c("#e5bd5b","#6b7398","#76b183","#d57d2c","#2a2d26","#f9d4cc","#3c634e","#ea68c3")
beta_q1n <- hilldiv2::hillpair(count_table_cov_size, q=1, metric="S")
betaq1p <- hilldiv2::hillpair(count_table_cov_size, q=1, tree = tree, metric ="S")
```

### Statistical analysis from EHI workflow
```{r beta_div_neutral_stats, comment="", echo=TRUE, message=FALSE, warning=FALSE}
sample_table_adonis <- all_metadata %>%
    filter(sample %in% labels(beta_q1n)) %>%
    arrange(sample) %>%
    # mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
    # select(sample,region) %>%
    select_if(~ length(unique(.)) > 1) %>% #remove columns with all-identical values
    column_to_rownames(var = "sample") %>%
    as.data.frame()

sample_table_adonis_p <- all_metadata %>%
    filter(sample %in% labels(betaq1p)) %>%
    arrange(sample) %>%
    # mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
    # select(sample,region) %>%
    select_if(~ length(unique(.)) > 1) %>% #remove columns with all-identical values
    column_to_rownames(var = "sample") %>%
    as.data.frame()
```
```{r}
    adonis2(formula=beta_q1n ~ region, data=sample_table_adonis[labels(beta_q1n),], permutations=999) %>%
            as.matrix() %>%
            kable()

    adonis2(formula=betaq1p ~ region, data=sample_table_adonis[labels(betaq1p),], permutations=999) %>%
            as.matrix() %>%
            kable()
```

#### NMDS:
```{r beta div NMDS, comment="", echo=FALSE, message=FALSE, warning=FALSE}
beta_q1n_nmds <- beta_q1n %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                dplyr::left_join(all_metadata, by = join_by(sample == sample))

betaq1p_nmds <- betaq1p %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                dplyr::left_join(all_metadata, by = join_by(sample == sample))
```
```{r beta div NMDS plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
group_n <- length(unique(beta_q1n_nmds$region))

group_p <- length(unique(betaq1p_nmds$region))

beta_q1n_nmds %>%
            group_by(region) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=region)) +
                scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Region"))

betaq1p_nmds %>%
            group_by(region) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=region)) +
                scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Region"))
```

### Stats second method, to check EHI workflow results
```{r Adonis_Region EHI, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# order data
m2 <- as.matrix(beta_q1n) 
sample_table_adonis_row <- rownames_to_column(sample_table_adonis, "Samples")
metadata_ord_2<- sample_table_adonis[order(match(sample_table_adonis_row$Sample,rownames(m2))),]

#Permutation test
ps.disper.neutral <- betadisper(beta_q1n, metadata_ord_2$region) 
permutest(ps.disper.neutral, pairwise = TRUE) 
# F Statistic: 26.96
# p-value: 0.001
# THEREFORE the data does NOT have equal multivariate dispersions 

# Adonis
adonis2(beta_q1n ~ region, data =metadata_ord_2, permutations = 999)
# A p-value of <0.05 is significant. The R2 value will tell you what amount of variation is based off the clustering variable. in this case the region. 
# p-value: 0.001
# R2: 0.28818
# THEREFORE there IS a significant difference between the beta diversity in Ittoq and Daneborg, and the location accounts for 22.6% of the difference 
```

Statistical results from EHI workflow match this workflow! 

# Functional diversity analysis analysis
```{r filter_annotations, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
#### read table ####
filtered_df <- read.table("all_annotations.csv", sep="")
```

## Functional GIFT distillation:
```{r gifts, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide', cache=TRUE}
count_filtered_rel <- count_filtered %>%
  rownames_to_column(., "Genome") %>%
  mutate_at(vars(-Genome),~./sum(.))  %>%
  column_to_rownames(., "Genome")

#Run distillation
GIFTs <- distill(filtered_df,GIFT_db,genomecol=2,annotcol=c(9,10,19))
#GIFTs <- GIFTs[-c(9,10),]
GIFT_db_table <- GIFT_db

# counts filtered correctly to the GIFTs distillation. Use this for counts from here on out. 
count_filtered_use_this <- count_filtered_rel[rownames(count_filtered_rel) %in% rownames(GIFTs),]

# Aggregate bundle-level GIFTs into the compound level 
# This is the level we will use for analysis
GIFTs_elements <- to.elements(GIFTs,GIFT_db)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

#Get community-weighed average GIFTs per sample
GIFTs_elements_community <- to.community(GIFTs_elements,count_filtered_use_this,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,count_filtered_use_this,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,count_filtered_use_this,GIFT_db)

#Convert traits into distance matrix
dist <- traits2dist(GIFTs_elements, method="gower")
```

#### Average functional diversities
```{r alpha_div_funct, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
alpha_div_func <- hilldiv(data=count_filtered_use_this,q=1,dist=dist)

#Q (Rao's Q); D_q (functional hill number, the effective number of equally abundant and functionally equally distinct species); MD_q (mean functional diversity per species, the effective sum of pairwise distances between a fixed species and all other species); FD_q (total functional diversity, the effective total functional distance between species of the assemblage).
```

```{r div_F_mean, comment="", echo=TRUE, message=FALSE, warning=FALSE}
alpha_div_F <-  t(alpha_div_func) %>%
  as.data.frame() %>%
  tibble::rownames_to_column("sample") %>%
  merge(., all_metadata, by="sample")

table_mean_alpha_func <- alpha_div_F %>%
  group_by(region) %>% 
  summarise_at(.vars = names(.)[2],.funs = c(mean="mean", sd="sd"))

knitr::kable(table_mean_alpha_func, format = "html", full_width = F,col.names = c('Groups', 'Mean', 'SD'), digits = 3) %>%
  kable_styling(latex_options="scale_down")

#Daneborg Mean 1.423; SD 0.034
#Ittoq Mean 1.453; SD 0.058
```

#### Functional diversity plots - Alpha diversity
```{r alpha_div_func_plot1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# Box plot for functional diversity
alpha_div_F %>%
  ggplot(aes(x = region, y = q1, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Functional Alpha diversity")
```

```{r alpha div functional stats, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Shapiro-Wilk test for normality
alpha_div_F %>%
  filter(region %in% c("Daneborg", "Ittoqqortoormii")) %>%
  group_by(region) %>%
  summarise(shapiro_p = shapiro.test(q1)$p.value) -> shapiro_results_F
# Results: p =  7.498381e-09	(Daneborg), p = 2.627117e-03	 Ittoqqortoormiit
# THEREFORE the data is NOT NORMALLY distributed


# Using leveneTest()
car:: leveneTest(q1 ~ region, alpha_div_F)
# Results: p-value = 0.03988 F-value = 4.4269
# THEREFORE therefore equal variance cannot be assumed


# Wilcoxon signed rank test for statistical significance
alpha_div_F %>%
  filter(region %in% c("Daneborg", "Ittoqqortoormii")) %>%
  group_by(region) %>%
  summarise(wilcox_p = wilcox.test(q1 ~ region, data = .)$p.value) -> wilcox_results_F
print(wilcox_results_F)
# Results: p = 0.0001328166	
# THEREFORE the difference in alpha diversity between Ittoq and Daneborg IS considered statistically significant. 
```

## Functional beta diversity calculations

```{r beta_div_func, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
beta_div_func <- hillpair(data=count_filtered_use_this,q=1,dist=dist)
beta_div_func_S <- beta_div_func$S
```

### Statistical analysis for func beta div using EHI Workflow:
```{r func beta div stats EHI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sample_table_adonis_func <- all_metadata %>%
    filter(sample %in% labels(beta_div_func_S)) %>%
    arrange(sample) %>%
    # mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
    # select(sample,region) %>%
    select_if(~ length(unique(.)) > 1) %>% #remove columns with all-identical values
    column_to_rownames(var = "sample") %>%
    as.data.frame()
```
```{r func beta div stats EHI table, comment="", echo=FALSE, message=FALSE, warning=FALSE}
adonis2(beta_div_func_S ~ region, data=sample_table_adonis_func[labels(beta_div_func_S),], permutations=999) %>%
            as.matrix() %>%
            kable()
```
```{r func beta div NMDS, comment="", echo=FALSE, message=FALSE, warning=FALSE}
beta_div_func_S_nmds <- beta_div_func_S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                dplyr::left_join(all_metadata, by = join_by(sample == sample))
```
```{r func beta div NMDS plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
group_func <- length(unique(beta_div_func_S_nmds$region))

beta_div_func_S_nmds %>%
            group_by(region) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=region)) +
                scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Region"))
```

### Statistical analysis using old workflow to check EHI
```{r Adonis_Region, comment="", echo=TRUE, message=FALSE, warning=FALSE}
# order data
m3 <- as.matrix(beta_div_func_S) 
sample_table_adonis_row_func <- rownames_to_column(sample_table_adonis_func, "Samples")
metadata_ord_3<- sample_table_adonis_func[order(match(sample_table_adonis_row_func$Sample,rownames(m2))),]


#Permutation test
ps.disper.func <- betadisper(beta_div_func_S, metadata_ord_3$region) 
permutest(ps.disper.func, pairwise = TRUE) 
# F Statistic: 7.9074
# p-value: 0.005
# THEREFORE the data does NOT have equal multivariate dispersions 
 

# Adonis
adonis2(beta_div_func_S ~ region, data =metadata_ord_3, permutations = 999)
# A p-value of <0.05 is significant.
# p-value: 0.954
# THEREFORE there IS NOT a significant difference between the functional beta diversity in Ittoq and Daneborg
```

## Functional capacity of the MAGs
```{r GIFTs_elements, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.dim = c(20, 20)}
GIFTs_elements %>%
  reshape2::melt() %>%
  dplyr::rename(Genome = Var1, Code_element = Var2, GIFT = value) %>%
  dplyr::inner_join(GIFT_db,by="Code_element") %>%
  ggplot(., aes(x=Code_element, y=Genome, fill=GIFT, group=Code_function))+
  geom_tile()+
  scale_y_discrete(guide = guide_axis(check.overlap = TRUE))+
  scale_x_discrete(guide = guide_axis(check.overlap = TRUE))+
  scale_fill_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  facet_grid(. ~ Code_function, scales = "free", space = "free")+
  theme_grey(base_size=8)+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),strip.text.x = element_text(angle = 90))
```
```{r hierachical_clustering, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.dim = c(20, 20)}
col <- colorRampPalette(brewer.pal(7, "YlGnBu"))(256)
heatmap(GIFTs_functions_community, scale = "none", col=col, Colv=NA,
        trace = "none", density.info = "none", ylab="Samples", 
        xlab="Gifts", cexRow=0.7, margins = c(8,8), lhei=c(2,4), lwid=c(2,5), 
        keysize=0.75, key.par = list(cex=0.9),srtCol = 45)
```


```{r tsne_prep, comment="", echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
taxonomy <- taxonomyclean1
```
```{r tsne, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(100)

GIFTs_elements_tSNE <- Rtsne(X=GIFTs_elements, dims = 2, perplexity = floor((nrow(GIFTs_elements) - 1) / 3), check_duplicates = FALSE)

GIFTs_domains <- as.data.frame(GIFTs_domains)
GIFTs_domains$genome <- rownames(GIFTs_domains)

GIFTs_elements_tSNE <- GIFTs_elements_tSNE$Y %>%
  as.data.frame() %>%
  mutate(genome=rownames(GIFTs_elements)) %>%
  dplyr::inner_join(GIFTs_domains, by="genome") %>%
  dplyr::inner_join(taxonomyclean, by="genome") %>%
  mutate_at(vars(phylum, class, order, family, genus), factor) %>%
  #mutate(cyl = factor(Phylum, levels = phylum_colors$Phylum)) %>%
  dplyr::rename(tSNE1="V1", tSNE2="V2")

count_t <- count_filtered_use_this %>%
  t() %>%
  as.data.frame()

#colnames(count_t) <- count_t[1,]
#count_t <- count_t[-1, ]
```

Create Merge table and GIFT elements for Ittoq dogs
```{r tsne_Ittoq, comment="", echo=FALSE, message=FALSE, warning=FALSE}
mergetable_Ittoq <- count_t %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column("sample") %>%
  merge(., all_metadata[ , c("sample", "region")], by="sample", all.x = TRUE) %>%
  filter(region=="Ittoqqortoormii") %>%
  column_to_rownames(., "sample")

mergetable_Ittoq <- mergetable_Ittoq [1: ncol(mergetable_Ittoq)-1 ]

GIFTs_elements_tSNE_rel_means_I <- colMeans(mergetable_Ittoq) %>%
  as.data.frame() %>%
  dplyr::rename( "Relative_value" = ".")  %>%
  rownames_to_column(., "genome") %>%
  merge(., GIFTs_elements_tSNE, by= "genome")

GIFTs_elements_tSNE_rel_means_Ittoq <- GIFTs_elements_tSNE_rel_means_I[GIFTs_elements_tSNE_rel_means_I$Relative_value != 0, ]

```

Create Merge table and GIFT elements for Daneborg dogs
```{r tsne_Daneborg, comment="", echo=FALSE, message=FALSE, warning=FALSE}
mergetable_Daneborg <- count_t %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column("sample") %>%
  merge(., all_metadata[ , c("sample", "region")], by="sample", all.x = TRUE) %>%
  filter(region=="Daneborg") %>%
  column_to_rownames(., "sample")

mergetable_Daneborg <- mergetable_Daneborg [1: ncol(mergetable_Daneborg)-1 ]

GIFTs_elements_tSNE_rel_means_D <- colMeans(mergetable_Daneborg) %>%
  as.data.frame() %>%
  dplyr::rename( "Relative_value" = ".")  %>%
  rownames_to_column(., "genome") %>%
  merge(., GIFTs_elements_tSNE, by= "genome")

GIFTs_elements_tSNE_rel_means_Daneborg <- GIFTs_elements_tSNE_rel_means_D[GIFTs_elements_tSNE_rel_means_D$Relative_value != 0, ]
```

### Region Daneborg: tSNE Phylum
```{r tsneIttoq_plot_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
GIFTs_elements_tSNE_rel_means_Ittoq %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = phylum))+
  geom_point(aes(size = Relative_value), shape=16, alpha=0.8) +
#  scale_color_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  theme_minimal() +
  theme()
```

### Region Daneborg: tSNE Phylum

```{r tsneDaneborg_plot_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
GIFTs_elements_tSNE_rel_means_Daneborg %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = phylum))+
  geom_point(aes(size = Relative_value), shape=16, alpha=0.8) +
  #scale_color_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  theme_minimal() +
  theme()
```

### Region Ittoq: tSNE gifts

```{r tsneIttoq_plot_gift, comment="", echo=FALSE, message=FALSE, warning=FALSE}
GIFTs_elements_tSNE_rel_means_Ittoq %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = Overall))+ #coloured by GIFT
  geom_point(aes(size = Relative_value), shape=16, alpha=0.8) +
  scale_color_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  theme_minimal() +
  theme()
```

### Region Daneborg: tSNE gifts

```{r tsneDaneborg_plot_gift, comment="", echo=FALSE, message=FALSE, warning=FALSE}
GIFTs_elements_tSNE_rel_means_Daneborg %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = Overall))+ 
  geom_point(aes(size = Relative_value), shape=16, alpha=0.8) +
  scale_color_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  theme_minimal() +
  theme()
```

# DA
```{r remove structural zeros}
# remove MAGs only present in one treatment, in this case the region
Daneborg_samples <- all_metadata %>% 
                    filter(region == "Daneborg") %>% 
                    dplyr::select(sample) %>% pull()

Ittoqqortoormii_samples <- all_metadata %>% 
                    filter(region == "Ittoqqortoormii") %>% 
                    dplyr::select(sample) %>% pull()

# count_table_cov_size <- rownames_to_column(count_table_cov_size, var = "genome") # only run once

structural_zeros <- count_table_cov_size %>% 
   dplyr::rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_Ittoq = all(c_across(all_of(Ittoqqortoormii_samples)) == 0)) %>% # set true if all samples in Ittoq have zeros
   mutate(all_zeros_Daneborg = all(c_across(all_of(Daneborg_samples)) == 0)) %>% # set true if all samples in Daneborg have zeros
   mutate(average_Ittoq = mean(c_across(all_of(Ittoqqortoormii_samples)), na.rm = TRUE)) %>% # get average genome counts across Ittoq
   mutate(average_Daneborg = mean(c_across(all_of(Daneborg_samples)), na.rm = TRUE)) %>% # get average genome counts across Daneborg
   filter(all_zeros_Ittoq == TRUE || all_zeros_Daneborg==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_Ittoq & !all_zeros_Daneborg ~ "Daneborg",
      !all_zeros_Ittoq & all_zeros_Daneborg ~ "Ittoq",
      !all_zeros_Ittoq & !all_zeros_Daneborg ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "Ittoq", average_Ittoq, average_Daneborg)) %>%
   dplyr::select(genome, present, average) %>%
   dplyr::left_join(mags_table, by=join_by(genome==genome)) %>%
   arrange(present,-average)
```
```{r plot structural zeros}
structural_zeros %>%
    mutate(average = ifelse(present == "Ittoq", average * -1, average)) %>% #convert Ittoq genome counts to negative
    ggplot(., aes(x=average, y=forcats::fct_rev(phylum), color=phylum)) +
      geom_jitter(size=3) +
      geom_vline(xintercept=0) + 
      xlim(-max(structural_zeros$average)-3,max(structural_zeros$average)+3) +
      scale_color_manual(values=phylum_colors) +
      geom_text(aes(-max(structural_zeros$average)+5, 1), label = "Only present\nin Ittoq", color="#666666") +
      geom_text(aes(max(structural_zeros$average)-5, 1), label = "Only present\nin Daneborg", color="#666666") +
     theme(legend.position='none',
          panel.background = element_blank(),
          axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"))+
      labs(y="Genus",x="Genome counts") + 
      guides(col=guide_legend("Phylum"))
```
```{r physeq object with normalized counts, structural zeros removed}
## Making phyloseq object -- for DA analysis
# OTU table
MAGotu <- count_table_cov_size %>% # genome size normalized counts
  dplyr::filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
  column_to_rownames("genome") %>%
  otu_table(., taxa_are_rows = TRUE)
# Tax table
taxonomyclean1 <- taxonomyclean[,-1]
taxmatrix <- as.matrix(taxonomyclean1[1:8])
rownames(taxmatrix) <- taxonomyclean$genome
taxtable <- tax_table(taxmatrix) 
#Metadata
metadata <- na.omit(metadata)
rownames(metadata) <- NULL
metadata.pre <- column_to_rownames(metadata, "sample")
#Sample table
sample_tab <- sample_data(metadata.pre)
# Tree
tree <- phytools::force.ultrametric(tree, method = "extend")
treephylo = phyloseq::phy_tree(tree)
## Merge into phyloseq object
physeq <- phyloseq(MAGotu, treephylo, sample_tab,taxtable)
```

## CLR transformation and wilcoxon testing
### Genome level
CLR transforming phyloseq object:
```{r clr transform for wilcoxon, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Perform CLR transformation
physeq_clr <- microbiome::transform(physeq, 'clr') # use whenever a package does NOT include its own data transformation (Wilcoxon)

# create regional groups
Daneborg_physeq <- subset_samples(physeq_clr, region == "Daneborg")
Ittoq_physeq <- subset_samples(physeq_clr, region == "Ittoqqortoormii")
```
Ittoq
```{r Ittoq summary for DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_ittoq <- subset_samples(physeq_clr, region == "Ittoqqortoormii")
physeq_ittoq <- prune_taxa(taxa_sums(physeq_ittoq)>0, physeq_ittoq)
table.rel1_ittoq <- physeq_ittoq@otu_table
means.table.rel1_ittoq <- as.data.frame(rowMeans(table.rel1_ittoq))
sd.table.rel1_ittoq <- as.data.frame(matrixStats::rowSds(table.rel1_ittoq, useNames = TRUE))
summary.ittoq <- merge(means.table.rel1_ittoq, sd.table.rel1_ittoq, by="row.names")
colnames(summary.ittoq) <- c("Genome","Mean", "SD")
print(summary.ittoq[order(-summary.ittoq$Mean),], row.names = FALSE)
```

Daneborg
```{r Daneborg summary for DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_daneborg <- subset_samples(physeq_clr, region == "Daneborg")
physeq_daneborg <- prune_taxa(taxa_sums(physeq_daneborg)>0, physeq_daneborg)
table.rel1_daneborg <- physeq_daneborg@otu_table
means.table.rel1_daneborg <- as.data.frame(rowMeans(table.rel1_daneborg))
sd.table.rel1_daneborg <- as.data.frame(matrixStats::rowSds(table.rel1_daneborg, useNames = TRUE))
summary.daneborg <- merge(means.table.rel1_daneborg, sd.table.rel1_daneborg, by="row.names")
colnames(summary.daneborg) <- c("Genome","Mean", "SD")
print(summary.daneborg[order(-summary.daneborg$Mean),], row.names = FALSE)
```

```{r mean_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr.t <- as.data.frame(t(as.matrix(physeq_clr@otu_table)))
table.W <- tibble::rownames_to_column(physeq_clr.t, "sample")
sample_type <- all_metadata[,c(1,4)]
table.W.meta <- merge(table.W,sample_type,by="sample")
table.W.meta <- table.W.meta[,-c(1)]
```
```{r wilcox_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr.t <- as.data.frame(t(as.matrix(physeq_clr@otu_table)))
table.W <- tibble::rownames_to_column(physeq_clr.t, "sample")
sample_type <- metadata[,c(1,4)]
genome_table <- merge(table.W,sample_type,by="sample")
genome_table <- genome_table[,-c(1)]
genome_table.no <- genome_table[,-ncol(genome_table)]
all.taxa <- colnames(genome_table.no)
Wilcox_result <- c()
for (y in all.taxa){
  res.wilcox <- wilcox.test(genome_table[,y] ~ region, data = genome_table,
                            exact = FALSE, alternative = "less")
  Wilcox_result <- rbind(Wilcox_result,c(res.wilcox[[1]],Pvalue=res.wilcox[[3]]))
}
rownames(Wilcox_result) <- all.taxa
Wilcox_result <- as.data.frame(Wilcox_result)

Wilcox_result$p_adjusted <- p.adjust(Wilcox_result$Pvalue, method = "fdr")

Wilcox_result_sign <- subset(Wilcox_result, p_adjusted <= 0.05) %>%
  rownames_to_column(., "genome")
knitr::kable(Wilcox_result_sign, format = "html", full_width = F,col.names = c("Genome", "W", "p-value","p-value adjusted"), digits = 10) %>%
  kable_styling(latex_options="scale_down")

Wilcox_result_sign <- Wilcox_result_sign %>%
    dplyr::left_join(taxonomyclean,by=join_by(genome==genome))
```

```{r selec_mean_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
table.W_filtered <- physeq_clr.t[colnames(physeq_clr.t) %in% Wilcox_result_sign$Genome]
sample_type <- all_metadata[,c(1,4)]
table.W_filtered_meta <- table.W_filtered %>%
  rownames_to_column(., "sample") %>%
  merge(.,sample_type,by="sample")
table.W_filtered_meta <- table.W_filtered_meta[,-c(1)]
means_genome_by_region <- table.W_filtered_meta %>% group_by(region) %>% summarise_at(.vars = names(.)[1:3],.funs = c(mean="mean", sd="sd"))
means <- as.data.frame(means_genome_by_region)
means <- means[,c(1,2,5,3,6,4,7)]
knitr::kable(means, format = "html", full_width = F, digits = 3) %>%
  kable_styling(latex_options="scale_down")
```
### Genus level
CLR transformation
```{r clr transform for DA G, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# agregate results at the genus level
physeq_clr_genus <- microbiome::aggregate_taxa(physeq_clr, 'genus')
```

Ittoq
```{r Ittoq summary for DA G, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_genus_ittoq <- subset_samples(physeq_clr_genus, region == "Ittoqqortoormii")
physeq_genus_ittoq <- prune_taxa(taxa_sums(physeq_genus_ittoq)>0, physeq_genus_ittoq)
table.rel1_ittoq <- physeq_genus_ittoq@otu_table
means.table.rel1_ittoq <- as.data.frame(rowMeans(table.rel1_ittoq))
sd.table.rel1_ittoq <- as.data.frame(matrixStats::rowSds(table.rel1_ittoq, useNames = TRUE))
summary.genus1_ittoq <- merge(means.table.rel1_ittoq, sd.table.rel1_ittoq, by="row.names")
colnames(summary.genus1_ittoq) <- c("genus","Mean", "SD")
print(summary.genus1_ittoq[order(-summary.genus1_ittoq$Mean),], row.names = FALSE)
```

Daneborg
```{r Daneborg summary for DA G, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_genus_daneborg <- subset_samples(physeq_clr_genus, region == "Daneborg")
physeq_genus_daneborg <- prune_taxa(taxa_sums(physeq_genus_daneborg)>0, physeq_genus_daneborg)
table.rel1_daneborg <- physeq_genus_daneborg@otu_table
means.table.rel1_daneborg <- as.data.frame(rowMeans(table.rel1_daneborg))
sd.table.rel1_daneborg <- as.data.frame(matrixStats::rowSds(table.rel1_daneborg, useNames = TRUE))
summary.genus1_daneborg <- merge(means.table.rel1_daneborg, sd.table.rel1_daneborg, by="row.names")
colnames(summary.genus1_daneborg) <- c("genus","Mean", "SD")
print(summary.genus1_daneborg[order(-summary.genus1_daneborg$Mean),], row.names = FALSE)
```

```{r mean_genus, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr_genus.t <- as.data.frame(t(as.matrix(physeq_clr_genus@otu_table)))
table.W_genus <- tibble::rownames_to_column(physeq_clr_genus.t, "sample")
sample_type <- all_metadata[,c(1,4)]
table.W.meta_genus <- merge(table.W_genus,sample_type,by="sample")
table.W.meta_genus <- table.W.meta_genus[,-c(1)]
```
```{r wilcox_genus, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr_genus.t <- as.data.frame(t(as.matrix(physeq_clr_genus@otu_table)))
table.W <- tibble::rownames_to_column(physeq_clr_genus.t, "sample")
sample_type_genus <- metadata[,c(1,4)]
genus_table <- merge(table.W,sample_type_genus,by="sample")
genus_table <- genus_table[,-c(1)]
genus_table.no <- genus_table[,-ncol(genus_table)]
all.taxa_genus <- colnames(genus_table.no)
Wilcox_result_genus <- c()
for (y in all.taxa_genus){
  res.wilcox <- wilcox.test(genus_table[,y] ~ region, data = genus_table,
                            exact = FALSE, alternative = "less")
  Wilcox_result_genus <- rbind(Wilcox_result_genus,c(res.wilcox[[1]],Pvalue=res.wilcox[[3]]))
}
rownames(Wilcox_result_genus) <- all.taxa_genus
Wilcox_result_genus <- as.data.frame(Wilcox_result_genus)

Wilcox_result_genus$p_adjusted <- p.adjust(Wilcox_result_genus$Pvalue, method = "fdr")

Wilcox_result_sign_genus <- subset(Wilcox_result_genus, p_adjusted <= 0.05) %>%
  rownames_to_column(., "Genus")
knitr::kable(Wilcox_result_sign_genus, format = "html", full_width = F,col.names = c("genus", "W", "p-value","p-value adjusted"), digits = 10) %>%
  kable_styling(latex_options="scale_down")
```

```{r selec_mean_genus, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr_genus.t <- as.data.frame(t(as.matrix(physeq_clr_genus@otu_table)))
table.W_genus <- tibble::rownames_to_column(physeq_clr_genus.t, "sample")
sample_type <- all_metadata[,c(1,4)]
table.W.meta_genus <- merge(table.W_genus,sample_type,by="sample")
table.W.meta_genus <- table.W.meta_genus[,-c(1)]
means_genus_by_region_all <- table.W.meta_genus %>% group_by(region) %>% summarise_at(.vars = names(.)[1:12],.funs = c(Maximum="max", Minimum="min",mean="mean", sd="sd"))
means_genus_by_region <- table.W.meta %>% group_by(region) %>% summarise_at(.vars = names(.)[1:12],.funs = c(mean="mean", sd="sd"))
means_all_genus <- as.data.frame(means_genus_by_region_all)
means <- as.data.frame(means_genus_by_region)

means_genus <- means[,c(1,2,14,3,15,4,16,5,17,6,18,7,19,8,20,9,21,10,22,11,23,12,24,13,25)]

knitr::kable(means_genus, format = "html", full_width = F, digits = 3) %>%
  kable_styling(latex_options="scale_down")
```
```{r}
# Collinsella
g1<- table.W.meta_genus %>%
  ggplot(aes(x = region, y = Collinsella, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Collinsella")
# Blautia_A
g2<- table.W.meta_genus %>%
  ggplot(aes(x = region, y = Blautia_A, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Blautia_A")
# Oliverpabstia
g3<- table.W.meta_genus %>%
  ggplot(aes(x = region, y = Oliverpabstia, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Oliverpabstia")
# Allobaculum
g4<- table.W.meta_genus %>%
  ggplot(aes(x = region, y = Allobaculum, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Allobaculum")
# Paraprevotella
g5<- table.W.meta_genus %>%
  ggplot(aes(x = region, y = Paraprevotella, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Paraprevotella")
# Mucispirillum
g6<- table.W.meta_genus %>%
  ggplot(aes(x = region, y = Bacteria_Deferribacterota_Deferribacteres_Deferribacterales_Mucispirillaceae_, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE, coef = 0) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  xlab("Region") + 
  ylab("Mucispirillum")

library(gridExtra)
grid.arrange(arrangeGrob(g1, g2,g3, g4, g5, g6, ncol = 3))
```
## DESeq2
### Genome level
```{r DESeq2 DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# convert phyloseq object to DESeq object
region_deseq <- phyloseq_to_deseq2(physeq, ~ region) # DESeq runs with its own transformation, do not use clr data
# run analysis
region_deseq <- DESeq(region_deseq)
results_table <- results(region_deseq)
```
```{r DESeq2 summary DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
deseq_res_ittoq_vs_daneborg <- results(region_deseq, alpha=0.05, contrast=c("region", "Ittoqqortoormii", "Daneborg"))
summary(deseq_res_ittoq_vs_daneborg) 
# limit to only show results significant at p- adjusted value of 0.05
sigtab_res_deseq_res_ittoq_vs_daneborg <- deseq_res_ittoq_vs_daneborg[which(deseq_res_ittoq_vs_daneborg$padj < 0.05), ]
summary(sigtab_res_deseq_res_ittoq_vs_daneborg) 
```
there are 111 MAGs enriched in ittoq compared to Daneborg, 62 showing decreased expression in ittoq compared to daneborg

Add in taxonomic information and sort by baseMean
```{r DESeq2 DA P baseMean, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sigtab_deseq <- deseq_res_ittoq_vs_daneborg[which(deseq_res_ittoq_vs_daneborg$padj < 0.05), ]

sigtab_deseq_region_with_tax <- cbind(as(sigtab_deseq, "data.frame"), as(tax_table(taxtable)[row.names(sigtab_deseq), ], "matrix"))
sigtab_deseq_region_with_tax[order(sigtab_deseq_region_with_tax$baseMean, decreasing=T), ]
sigtab_deseq_region_with_tax$genome <- rownames(sigtab_deseq_region_with_tax)

deseq2_ins_fis <- as.data.frame(sigtab_deseq_region_with_tax)

x = tapply(sigtab_deseq_region_with_tax$log2FoldChange, sigtab_deseq_region_with_tax$genome, function(x) max(x))
x = sort(x, TRUE)
sigtab_deseq_region_with_tax$genome = factor(as.character(sigtab_deseq_region_with_tax$genome), levels=names(x))
x = tapply(sigtab_deseq_region_with_tax$log2FoldChange, sigtab_deseq_region_with_tax$genome, function(x) max(x))
x = sort(x, TRUE)
sigtab_deseq_region_with_tax$genome = factor(as.character(sigtab_deseq_region_with_tax$genome), levels=names(x))

colors_alphabetic <- ehi_phylum_colors1 %>%
  dplyr::right_join(taxonomyclean, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
	unique() %>%
	dplyr::arrange(phylum)

sigtab_deseq_with_tax1 <- sigtab_deseq_region_with_tax %>%	dplyr::arrange(phylum)
rownames(sigtab_deseq_with_tax1) <- sigtab_deseq_with_tax1$genome
```

```{r DESeq2 DA P plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ggplot(sigtab_deseq_with_tax1, aes(x=forcats::fct_rev(reorder(genome, as.integer(factor(genome)))), y=log2FoldChange, color=phylum)) + 
  geom_point(size=4) + 
 scale_color_manual(values=colors_alphabetic$colors) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("Genome") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
### Genus level
```{r DESeq2 DA G, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# aggregate at genus level
physeq.genus <- tax_glom(physeq, 'genus')

region_deseq_genus <- phyloseq_to_deseq2(physeq.genus, ~ region) 

region_deseq_genus <- DESeq(region_deseq_genus)
results_table <- results(region_deseq_genus)
```
```{r DESeq2 DA G summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
deseq_res_ittoq_vs_daneborg_genus <- results(region_deseq_genus, alpha=0.05, contrast=c("region", "Ittoqqortoormii", "Daneborg"))

summary(deseq_res_ittoq_vs_daneborg_genus) 

sigtab_res_deseq_res_ittoq_vs_daneborg_genus <- deseq_res_ittoq_vs_daneborg_genus[which(deseq_res_ittoq_vs_daneborg_genus$padj < 0.05), ]

summary(sigtab_res_deseq_res_ittoq_vs_daneborg_genus) 
```
34 genera enriched in Ittoq, 8 in Daneborg
```{r DESeq2 DA G stats, comment="", echo=FALSE, message=FALSE, warning=FALSE}
sigtab_deseq_genus <- deseq_res_ittoq_vs_daneborg_genus[which(deseq_res_ittoq_vs_daneborg_genus$padj < 0.05), ]

sigtab_deseq_region_genus_with_tax <- cbind(as(sigtab_deseq_genus, "data.frame"), as(tax_table(taxtable)[row.names(sigtab_deseq_genus), ], "matrix"))
sigtab_deseq_region_genus_with_tax[order(sigtab_deseq_region_genus_with_tax$baseMean, decreasing=T), ]

deseq2_ins_fis_2 <- as.data.frame(sigtab_deseq_region_genus_with_tax)

x = tapply(sigtab_deseq_region_genus_with_tax$log2FoldChange, sigtab_deseq_region_genus_with_tax$phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_deseq_region_genus_with_tax$phylum = factor(as.character(sigtab_deseq_region_genus_with_tax$phylum), levels=names(x))
x = tapply(sigtab_deseq_region_genus_with_tax$log2FoldChange, sigtab_deseq_region_genus_with_tax$genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_deseq_region_genus_with_tax$genus = factor(as.character(sigtab_deseq_region_genus_with_tax$genus), levels=names(x))

sigtab_deseq_with_tax2 <- sigtab_deseq_region_genus_with_tax %>%
  dplyr::select(genus, phylum, padj, log2FoldChange) %>%
  arrange(phylum)

rownames(sigtab_deseq_with_tax2) <- sigtab_deseq_with_tax2$genus
```
```{r DESeq2 DA G plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ggplot(sigtab_deseq_with_tax2, aes(x=forcats::fct_rev(reorder(genus, as.integer(factor(genus)))), y=log2FoldChange, color=phylum)) + 
  geom_point(size=4) + 
 scale_color_manual(values=colors_alphabetic$colors) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("Genus") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```

## ANCOM-BC
```{r}
phylo_samples <- all_metadata %>% 
                    column_to_rownames("sample") %>% 
                    mutate(region = factor(region, levels = c("Daneborg","Ittoqqortoormii"))) %>% #TJ1 = Daneborg, TJ2 = Ittoq
                    sample_data() #convert to phyloseq sample_data object

phylo_counts <- count_table_cov_size %>% 
                    filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
                    column_to_rownames("genome") %>% 
                    mutate_all(~ replace(., . == 0, 0.00001)) %>% #add pseudo counts to avoid structural zero issues 
                    otu_table(., taxa_are_rows = TRUE)

phylo_taxonomy <- taxonomyclean %>% 
                    filter(genome %in% rownames(phylo_counts)) %>% # remove structural zeros
                    mutate(genome2=genome) %>% #create a pseudo genome name column
                    column_to_rownames("genome2") %>% 
                    dplyr::select(domain,phylum,class,order,family,genus,species,genome) %>% # include all taxonomic information to ensure genome-level analysis
                    as.matrix() %>% 
                    tax_table()

#Generate phyloseq object required to input ANCOMBC, only difference between this phyloseq object and the one for DESeq is the addition of the psudocounts
physeq_ancom <- phyloseq(phylo_counts, phylo_taxonomy, phylo_samples)
```

### Genome level
```{r ANCOM-BC DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_output <- ancombc2(physeq_ancom, 
                  assay_name = "counts", 
                  tax_level = NULL, # no defined tax level
                  fix_formula = "region",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                  lib_cut = 1000, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = NULL, 
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

tax <- data.frame(physeq_ancom@tax_table) %>%
  rownames_to_column(., "taxon") 

ancom_result <- ancom_output$res %>%
    dplyr::rename(genome=taxon) %>%
    dplyr::left_join(taxonomyclean,by=join_by(genome==genome))

ancom_result %>%
    mutate(significance = ifelse(q_regionIttoqqortoormii < 0.05, "1", "0")) %>%
    ggplot(., aes(x=-log(p_regionIttoqqortoormii), y=lfc_regionIttoqqortoormii, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      geom_text(aes(2.5, 10), label = "Enriched\nin Ittoq", color="#666666") +
      geom_text(aes(2.5, -10), label = "Enriched\nin Daneborg", color="#666666") +
      labs(color="Significance", y="Difference between treatments", x="p-value") +
      theme_classic()
```
```{r}
ancom_result %>%
    filter(q_regionIttoqqortoormii < 0.05) %>%
    ggplot(., aes(x=lfc_regionIttoqqortoormii, y=forcats::fct_rev(genus), color=phylum)) +
      geom_jitter(size=3) +
      geom_vline(xintercept=0) + 
      xlim(-max(ancom_result$lfc_regionIttoqqortoormii)-2,max(ancom_result$lfc_regionIttoqqortoormii)+2) +
      scale_color_manual(values=colors_alphabetic$colors) +
      geom_text(aes(-max(ancom_result$lfc_regionIttoqqortoormii)+1, 1), label = "Enriched\nin Daneborg", color="#666666") +
      geom_text(aes(max(ancom_result$lfc_regionIttoqqortoormii)-1, 1), label = "Enriched\nin Ittoq", color="#666666") +
     theme(legend.position='none',
          panel.background = element_blank(),
          axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
          axis.title.x=element_blank())+
      labs(y="Phylum",x="Genome counts") + 
      guides(col=guide_legend("Phylum"))
```
```{r}
#Enriched Daneborg
ancom_result %>%
  filter(lfc_regionIttoqqortoormii < 0) %>% 
  filter(q_regionIttoqqortoormii < 0.05) %>% # using q (adjusted p-value)
  arrange(lfc_regionIttoqqortoormii) %>%
  dplyr::select(genome,lfc_regionIttoqqortoormii,q_regionIttoqqortoormii) %>%
  kable()

# Enriched Ittoq
ancom_result %>%
  filter(lfc_regionIttoqqortoormii > 0) %>% 
  filter(q_regionIttoqqortoormii < 0.05) %>% 
  arrange(-lfc_regionIttoqqortoormii) %>%
  dplyr::select(genome,lfc_regionIttoqqortoormii,q_regionIttoqqortoormii) %>%
  kable()
```
```{r ANCOM-BC DA P plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_result %>%
  ggplot(., aes(x=forcats::fct_rev(genus), y=lfc_regionIttoqqortoormii, color=phylum)) + 
  geom_point(size=4) + 
  scale_color_manual(values=colors_alphabetic$colors) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("phylum") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
### Genus level
```{r ANCOM-BC DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_output_genus <- ancombc2(physeq_ancom, 
                  assay_name = "counts", 
                  tax_level = "genus", # limit calculations to genus level
                  fix_formula = "region",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0.10, 
                  lib_cut = 1000, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = NULL, 
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

tax <- data.frame(physeq_ancom@tax_table) %>%
  rownames_to_column(., "taxon") 

ancom_result_genus <- ancom_output_genus$res %>%
    dplyr::rename(genome=taxon) %>%
    dplyr::left_join(taxonomyclean,by=join_by(genome==genome))

ancom_result_genus %>%
    mutate(significance = ifelse(q_regionIttoqqortoormii < 0.05, "1", "0")) %>%
    ggplot(., aes(x=-log(p_regionIttoqqortoormii), y=lfc_regionIttoqqortoormii, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      geom_text(aes(2.5, 10), label = "Enriched\nin Ittoq", color="#666666") +
      geom_text(aes(2.5, -10), label = "Enriched\nin Daneborg", color="#666666") +
      labs(color="Significance", y="Difference between treatments", x="p-value") +
      theme_classic()
```
```{r}
ancom_result_genus %>%
    filter(q_regionIttoqqortoormii < 0.05) %>%
    ggplot(., aes(x=lfc_regionIttoqqortoormii, y=forcats::fct_rev(genus), color=phylum)) +
      geom_jitter(size=3) +
      geom_vline(xintercept=0) + 
      xlim(-max(ancom_result_genus$lfc_regionIttoqqortoormii)-2,max(ancom_result_genus$lfc_regionIttoqqortoormii)+2) +
      scale_color_manual(values=colors_alphabetic$colors) +
      geom_text(aes(-max(ancom_result_genus$lfc_regionIttoqqortoormii)+1, 1), label = "Enriched\nin Daneborg", color="#666666") +
      geom_text(aes(max(ancom_result_genus$lfc_regionIttoqqortoormii)-1, 1), label = "Enriched\nin Ittoq", color="#666666") +
     theme(legend.position='none',
          panel.background = element_blank(),
          axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
          axis.title.x=element_blank())+
      labs(y="Phylum",x="Genome counts") + 
      guides(col=guide_legend("Phylum"))
```
```{r}
#Enriched Daneborg
ancom_result_genus %>%
  filter(lfc_regionIttoqqortoormii < 0) %>% 
  filter(q_regionIttoqqortoormii < 0.05) %>% # note use of q here
  arrange(lfc_regionIttoqqortoormii) %>%
  dplyr::select(genome,lfc_regionIttoqqortoormii,q_regionIttoqqortoormii) %>%
  kable()

# Enriched Ittoq
ancom_result_genus %>%
  filter(lfc_regionIttoqqortoormii > 0) %>% 
  filter(q_regionIttoqqortoormii < 0.05) %>% 
  arrange(-lfc_regionIttoqqortoormii) %>%
  dplyr::select(genome,lfc_regionIttoqqortoormii,q_regionIttoqqortoormii) %>%
  kable()

## need to figure out column naming and why it isnt working.
```
```{r ANCOM-BC DA P plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_result_genus %>%
  filter(q_regionIttoqqortoormii < 0.05) %>% 
  ggplot(., aes(x=forcats::fct_rev(genome), y=lfc_regionIttoqqortoormii, color=phylum)) + 
  geom_point(size=4) + 
  scale_color_manual(values=colors_alphabetic$colors) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("phylum") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```

### ADLEx2
```{r}
library(ALDEx2)

genome_counts_ALDEx <- count_table_cov_size %>%
  filter(!genome %in% structural_zeros$genome) %>% # remove structural zeros
  column_to_rownames(var="genome") %>%
  mutate_all(~ . * 1e6) %>% #multiply by a million
  round(0) #round to integer

count_ALDEx.clr <- aldex.clr(genome_counts_ALDEx, 
               all_metadata$region, 
               mc.samples=128, 
               denom="all", 
               verbose=F)

```
```{r}
genome_counts_ALDEx.ttest <- aldex.ttest(count_ALDEx.clr, 
                hist.plot=F, 
                paired.test=F, 
                verbose=F)

genome_counts_ALDEx.effect <- aldex.effect(count_ALDEx.clr, 
              CI=T, 
              verbose=F, 
              include.sample.summary=F, 
              glm.conds=NULL, 
              useMC=F)
```
```{r}
count_ALDEx.all <- data.frame(genome_counts_ALDEx.ttest,genome_counts_ALDEx.effect) %>%
    rownames_to_column(var="genome") %>%
    dplyr::left_join(taxonomyclean,by=join_by(genome==genome))
```
Bland-Altman plot
```{r}
count_ALDEx.all %>%
    mutate(significance = ifelse(we.eBH < 0.05, "1", "0")) %>%
    ggplot(., aes(x=rab.all, y=diff.btw, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      labs(color="Significance", y="Difference between treatments", x="Mean CLR") +
      theme_classic()
```
Volcano plot
```{r}
count_ALDEx.all %>%
    mutate(significance = ifelse(wi.eBH < 0.05, "1", "0")) %>%
    ggplot(., aes(x=-log(wi.eBH), y=diff.btw, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      labs(color="Significance", y="Difference between treatments", x="p-value") +
      theme_classic()
```
Effect-size plot
```{r}
count_ALDEx.all %>%
    mutate(significance = ifelse(wi.eBH < 0.05, "1", "0")) %>%
    ggplot(., aes(x=diff.win, y=diff.btw, color=significance)) +
      geom_abline(intercept = 0, slope =  0, size=0.4, linetype="solid",  color="#000000") +
      geom_abline(intercept = 0, slope =  1, size=0.4, linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope =  2, size=0.6,   linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope =  4, size=0.8,   linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope = -1, size=0.4, linetype="dashed", color="#000000") + 
      geom_abline(intercept = 0, slope = -2, size=0.6,   linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope = -4, size=0.8,   linetype="dashed", color="#000000") +
      geom_point() +
      scale_color_manual(values = c("#cccccc90","#00FFFF90")) +
      geom_text(aes(2.5, 20), label = "Enriched\nin Ittoq", color="#666666") +
      geom_text(aes(2.5, -20), label = "Enriched\nin Daneborg", color="#666666") +
      labs(color="Significance", y="Difference between treatments", x="Dispersion within treatments") +
      theme_classic()
```
Significance based on posterior probabilities
```{r}
count_ALDEx.all %>%
    mutate(significance = ifelse(overlap < 0.1, "1", "0")) %>%
    ggplot(., aes(x=diff.win, y=diff.btw, color=significance)) +
      geom_abline(intercept = 0, slope =  0, size=0.4, linetype="solid",  color="#000000") +
      geom_abline(intercept = 0, slope =  1, size=0.4, linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope =  2, size=0.6,   linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope =  4, size=0.8,   linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope = -1, size=0.4, linetype="dashed", color="#000000") + 
      geom_abline(intercept = 0, slope = -2, size=0.6,   linetype="dashed", color="#000000") +
      geom_abline(intercept = 0, slope = -4, size=0.8,   linetype="dashed", color="#000000") +
      geom_point() +
      scale_color_manual(values = c("#cccccc90","#00FFFF90")) +
      geom_text(aes(2.5, 20), label = "Enriched\nin Ittoq", color="#666666") +
      geom_text(aes(2.5, -20), label = "Enriched\nin Daneborg", color="#666666") +
      labs(color="Significance", y="Difference between treatments", x="Dispersion within treatments") +
      theme_classic()
```
Largest effect-size genomes
```{r}
count_ALDEx.all %>%
    filter(overlap < 0.5) %>% # changed to see if chart worked. tutorial has set at 0.1
    ggplot(., aes(x=effect, y=forcats::fct_rev(phylum), color=phylum)) +
      geom_jitter(size=3) +
      geom_vline(xintercept=0) + 
      xlim(-max(count_ALDEx.all$effect)-1,max(count_ALDEx.all$effect)+1) +
      scale_color_manual(values=colors_alphabetic$colors) +
      geom_text(aes(-max(count_ALDEx.all$effect)+1, 1), label = "Enriched\nin D", color="#666666") +
      geom_text(aes(max(count_ALDEx.all$effect)-1, 1), label = "Enriched\nin I", color="#666666") +
     theme(legend.position='none',
          panel.background = element_blank(),
          axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
          axis.title.x=element_blank())+
      labs(y="Genus",x="Genome counts") + 
      guides(col=guide_legend("Phylum"))
```
Top enriched in Ittoq
```{r}
# Enriched in Daneborg
count_ALDEx.all %>%
  filter(effect < -1) %>% #
  filter(overlap < 0.1) %>% # >90% of the estimated effect sizes are >0
  arrange(effect) %>%
  dplyr::select(genome,effect,phylum,order,genus) %>%
  kable()

# Enriched in Ittoq
count_ALDEx.all %>%
  filter(effect > 1) %>%
  filter(overlap < 0.1) %>% # >90% of the estimated effect sizes are >0
  arrange(-effect) %>%
  dplyr::select(genome,effect,phylum,order,genus) %>%
  kable()
```
### LinDA
#### Genome level
```{r LinDA DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tse_LinDA = mia::makeTreeSummarizedExperimentFromPhyloseq(physeq) # create tse object from the phyloseq object with structural zeros removed, no pseudo counts. 
tse_LinDA <- mia::subsetByPrevalentTaxa(tse_LinDA, detection = 0, prevalence = 0.1) # filter tse object
```
```{r LinDA DA P analysis, comment="", echo=FALSE, message=FALSE, warning=FALSE}
otu.tab <- as.data.frame(assay(tse_LinDA))
meta <- as.data.frame(colData(tse_LinDA)) %>% dplyr::select(region)
LinDA <- LinDA::linda(
  otu.tab, 
  meta, 
  formula = '~region', 
  alpha = 0.05, 
  prev.cut = 0, # we already filtered 
  lib.cut = 1000, 
  winsor.quan = 0.97)

LinDA
LinDA::linda.plot(LinDA, c('region') )

LinDA_table <- rownames_to_column(LinDA$output$regionIttoqqortoormii, "genome") 
```
```{r LinDA DA P summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq@tax_table) %>%
  rownames_to_column(., "taxon")

LinDA_table <- rownames_to_column(LinDA$output$regionIttoqqortoormii, "genome") 
LinDA_table <- merge(LinDA_table, taxonomyclean[, c("genome", "phylum", "genus", "species")], by = "genome", all.x = TRUE)

LinDA_table <- LinDA_table %>%
  dplyr::select(genome, phylum, genus, species,lfcSE, padj, log2FoldChange) %>%
  filter(padj < 0.005)

```
```{r LinDA DA P plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ggplot(LinDA_table, aes(x=forcats::fct_rev(genus), y=log2FoldChange, color=phylum)) + 
  geom_point(size=4) + 
  scale_color_manual(values=colors_alphabetic$colors) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("genome") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
#### genus level
```{r LinDA DA G, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq.genus_LinDA <- tax_glom(physeq, 'genus')
tse_LinDA_genus = mia::makeTreeSummarizedExperimentFromPhyloseq(physeq.genus_LinDA) # create tse object
tse_LinDA_genus <- mia::subsetByPrevalentTaxa(tse_LinDA_genus, detection = 0, prevalence = 0.1) # filter tse object
```
```{r LinDA DA G analysis, comment="", echo=FALSE, message=FALSE, warning=FALSE}
otu.tab_genus <- as.data.frame(assay(tse_LinDA_genus))
meta_genus <- as.data.frame(colData(tse_LinDA_genus)) %>% dplyr::select(region)
LinDA_genus <- LinDA::linda(
  otu.tab_genus, 
  meta_genus, 
  formula = '~region', 
  alpha = 0.05, 
  prev.cut = 0, # we already filtered 
  lib.cut = 1000, 
  winsor.quan = 0.97)

LinDA_genus
LinDA::linda.plot(LinDA_genus, c('region') )

LinDA_table_genus <- rownames_to_column(LinDA_genus$output$regionIttoqqortoormii, "genus") 
```
```{r LinDA DA G summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq.genus_LinDA@tax_table) %>%
  rownames_to_column(., "taxon")

LinDA_table_genus <- rownames_to_column(LinDA_genus$output$regionIttoqqortoormii, "genome") 
LinDA_table_genus <- merge(LinDA_table_genus, taxonomyclean[, c("genome", "phylum", "genus")], by = "genome", all.x = TRUE)

LinDA_table_genus <- LinDA_table_genus %>%
  dplyr::select(phylum, lfcSE, padj, log2FoldChange, genus) %>%
  filter(padj < 0.05)
```
```{r LinDA DA G plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ggplot(LinDA_table_genus, aes(x=forcats::fct_rev(genus), y=log2FoldChange, color=phylum)) + 
  geom_point(size=4) + 
  scale_color_manual(values=colors_alphabetic$colors) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("genus") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
#### Merged DA
```{r merged genome}
Wilcoxon_genome <- Wilcox_result_sign %>%
  dplyr::select(., genome, p_adjusted) %>%
  dplyr::rename(
    padj_WILCOX = p_adjusted,
    )
DESeq_genome <- sigtab_deseq_region_with_tax %>%
  dplyr::select(., genome, padj, log2FoldChange) %>%
  dplyr::rename(
    padj_DESEQ = padj,
    LogFold_DESEQ = log2FoldChange,
    )
ANCOMBC_genome <- ancom_result %>%
  filter(q_regionIttoqqortoormii < 0.05) %>%
  dplyr::select(., genome, q_regionIttoqqortoormii, lfc_regionIttoqqortoormii) %>%
  dplyr::rename(
    padj_ANCOMBC = q_regionIttoqqortoormii,
    LogFold_ANCOMBC = lfc_regionIttoqqortoormii,
    )
LinDA_genome <- LinDA_table  %>%
  filter(padj < 0.05) %>%
  dplyr::select(., genome, padj, log2FoldChange)  %>%
  dplyr::rename(
    padj_LINDA = padj,
    LogFold_LINDA = log2FoldChange,
    )
taxonomyclean_genome <- taxonomyclean %>%
  dplyr::select(., phylum, class, order, family, genus, species, genome)
  
DA_results_list <- list(Wilcoxon_genome, DESeq_genome, ANCOMBC_genome, LinDA_genome)
DA_results<- Reduce(function(x, y) merge(x, y, all=TRUE), DA_results_list) 
DA_results <- dplyr::left_join(DA_results, taxonomyclean_genome, by = "genome")

# library("writexl")
# write_xlsx(DA_results, "/Users/elsa/Desktop/THESIS/sled_dog_parasite\\DA_genome_tax.xlsx")
```
```{r merged genus}
Wilcoxon_genus <- Wilcox_result_sign_genus %>%
  dplyr::select(., Genus, p_adjusted) %>%
  dplyr::rename(
    padj_WILCOX = p_adjusted,
    genus = Genus
    )

DESeq_genus <- sigtab_deseq_with_tax2 %>%
  dplyr::select(., genus, padj, log2FoldChange) %>%
  dplyr::rename(
    padj_DESEQ = padj,
    LogFold_DESEQ = log2FoldChange,
    )
ancom_result_genus$genome <- sub("genus:", "", ancom_result_genus$genome)
ANCOMBC_genus <- ancom_result_genus %>%
  filter(q_regionIttoqqortoormii < 0.05) %>%
  dplyr::select(., genome, q_regionIttoqqortoormii, lfc_regionIttoqqortoormii) %>%
  dplyr::rename(
    padj_ANCOMBC = q_regionIttoqqortoormii,
    LogFold_ANCOMBC = lfc_regionIttoqqortoormii,
    genus = genome
    )
LinDA_genus <- LinDA_table_genus  %>%
  filter(padj < 0.05) %>%
  dplyr::select(., genus, padj, log2FoldChange)  %>%
  dplyr::rename(
    padj_LINDA = padj,
    LogFold_LINDA = log2FoldChange,
    )
taxonomyclean_genus <- taxonomyclean %>%
  dplyr::select(., phylum, class, order, family, genus)
  
DA_results_list_genus <- list(Wilcoxon_genus, DESeq_genus, ANCOMBC_genus, LinDA_genus)
DA_results_genus <- Reduce(function(x, y) merge(x, y, all=TRUE), DA_results_list_genus) 
DA_results_genus <- dplyr::left_join(DA_results_genus, taxonomyclean_genus, by = "genus")
DA_results_genus <- DA_results_genus[!duplicated(DA_results_genus), ]

# library("writexl")
# write_xlsx(DA_results_genus, "/Users/elsa/Desktop/THESIS/sled_dog_parasite\\DA_genus4.xlsx")
```
# Functional Capacity Differences
Kegg data prep from EHI workflow
```{r kegg data prep from EHI, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Microbiome functional attribute table
kegg_file="/Users/elsa/Desktop/THESIS/Thesis/DMB0032_merged_kegg.tsv"
kegg_table <- read.table(kegg_file,sep="\t",header=T)
kegg_table_clean <- dplyr::distinct(kegg_table) #remove duplicate rows
kegg_table_clean_1 <- count_filtered_rel[rownames(count_filtered_rel) %in% rownames(kegg_table_clean),]
```
# Functional capacity
## Community Domain level
```{r}
GIFTs_domains_community_df <- as.data.frame(GIFTs_domains_community)
GIFTs_domains_community_df$sample <- rownames(GIFTs_domains_community_df)

selected_columns_func <- all_metadata %>%
  dplyr::select(sample, region)

GIFTs_domains_community_df <- dplyr::left_join(GIFTs_domains_community_df, selected_columns_func, by = "sample")
```
```{r}
GIFTs_domains_community_df <- as.data.frame(GIFTs_domains_community) %>%
  rownames_to_column(., var = "sample")
GIFTs_domains_community_df <- merge(selected_columns_func, GIFTs_domains_community_df, by = "sample")
GIFTs_domains_community_df <- column_to_rownames(GIFTs_domains_community_df, var = "sample")
table.W_domain <- tibble::rownames_to_column(GIFTs_domains_community_df, "sample")
comm_domain_table <- subset(table.W_domain, select = -c(sample))
comm_domain_table.no <- subset(comm_domain_table, select = -c(region))
all.comm_domain <- colnames(comm_domain_table.no)

Wilcox_result_comm_domain <- c()
for (y in all.comm_domain){ 
  res.wilcox_comm_domain <- wilcox.test(comm_domain_table[,y] ~ region, data = comm_domain_table,
                            exact = FALSE, alternative = "less")
  Wilcox_result_comm_domain <- rbind(Wilcox_result_comm_domain,c(res.wilcox_comm_domain[[1]],Pvalue=res.wilcox_comm_domain[[3]]))
}
rownames(Wilcox_result_comm_domain) <- all.comm_domain
Wilcox_result_comm_domain <- as.data.frame(Wilcox_result_comm_domain)

Wilcox_result_comm_domain$p_adjusted <- p.adjust(Wilcox_result_comm_domain$Pvalue, method = "fdr")

Wilcox_result_comm_domain_sign <- subset(Wilcox_result_comm_domain, p_adjusted <= 0.05) %>%
  rownames_to_column(., "Domain")
knitr::kable(Wilcox_result_comm_domain_sign, format = "html", full_width = F,col.names = c("Domain", "W", "p-value","p-value adjusted"), digits = 10) %>%
  kable_styling(latex_options="scale_down")
```
Now at the Community Functions level:
```{r}
GIFTs_functions_community_df <- as.data.frame(GIFTs_functions_community) %>%
  rownames_to_column(., var = "sample")
GIFTs_functions_community_df <- merge(selected_columns_func, GIFTs_functions_community_df, by = "sample")
GIFTs_functions_community_df <- column_to_rownames(GIFTs_functions_community_df, var = "sample")
table.W_comm <- tibble::rownames_to_column(GIFTs_functions_community_df, "sample")
comm_func_table <- subset(table.W_comm, select = -c(sample))
comm_func_table.no <- subset(comm_func_table, select = -c(region))
all.comm_func <- colnames(comm_func_table.no)

Wilcox_result_comm_func <- c()
for (y in all.comm_func){ 
  res.wilcox_comm_func <- wilcox.test(comm_func_table[,y] ~ region, data = comm_func_table,
                            exact = FALSE, alternative = "less")
  Wilcox_result_comm_func <- rbind(Wilcox_result_comm_func,c(res.wilcox_comm_func[[1]],Pvalue=res.wilcox_comm_func[[3]]))
}
rownames(Wilcox_result_comm_func) <- all.comm_func
Wilcox_result_comm_func <- as.data.frame(Wilcox_result_comm_func)

Wilcox_result_comm_func$p_adjusted <- p.adjust(Wilcox_result_comm_func$Pvalue, method = "fdr")

Wilcox_result_comm_func_sign <- subset(Wilcox_result_comm_func, p_adjusted <= 0.05) %>%
  rownames_to_column(., "Code_function")
knitr::kable(Wilcox_result_comm_func_sign, format = "html", full_width = F,col.names = c("Code_function", "W", "p-value","p-value adjusted"), digits = 10) %>%
  kable_styling(latex_options="scale_down")

summary_table.W_comm <- table.W_comm %>%
  group_by(region) %>%
  summarise(
    mean_D09 = mean(D09),
    sd_D09 = sd(D09),
    mean_S02 = mean(S02),
    sd_S02 = sd(S02),
  ) %>%
  filter(region %in% regions)
```
```{r}
# D09-Antibiotic degradation
cf1<- ggplot(GIFTs_functions_community_df, aes(x = region, y = D09, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D09-Antibiotic degradation",
    x = "Region",
    y = "D09"
  )
# S02-Appendages
cf2<- ggplot(GIFTs_functions_community_df, aes(x = region, y = S02, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "S02-Appendages",
    x = "Region",
    y = "S02"
  )
# joint figure
library(gridExtra)
grid.arrange(arrangeGrob(cf1, cf2, ncol = 2))
```
#Community Element
```{r}
# community element level
GIFTs_elements_community_df <- as.data.frame(GIFTs_elements_community) %>%
  rownames_to_column(., var = "sample")
GIFTs_elements_community_df <- merge(selected_columns_func, GIFTs_elements_community_df, by = "sample")
GIFTs_elements_community_df <- column_to_rownames(GIFTs_elements_community_df, var = "sample")
table.W_elements <- tibble::rownames_to_column(GIFTs_elements_community_df, "sample")
comm_elements_table <- subset(table.W_elements, select = -c(sample))
comm_elements_table.no <- subset(comm_elements_table, select = -c(region))
all.comm_elements <- colnames(comm_elements_table.no)

Wilcox_result_comm_elements <- c()
for (y in all.comm_elements){ 
  res.wilcox_comm_elements <- wilcox.test(comm_elements_table[,y] ~ region, data = comm_elements_table,
                            exact = FALSE, alternative = "less")
  Wilcox_result_comm_elements <- rbind(Wilcox_result_comm_elements,c(res.wilcox_comm_elements[[1]],Pvalue=res.wilcox_comm_elements[[3]]))
}
rownames(Wilcox_result_comm_elements) <- all.comm_elements
Wilcox_result_comm_elements <- as.data.frame(Wilcox_result_comm_elements)

Wilcox_result_comm_elements$p_adjusted <- p.adjust(Wilcox_result_comm_elements$Pvalue, method = "fdr")

Wilcox_result_comm_elements_sign <- subset(Wilcox_result_comm_elements, p_adjusted <= 0.05) %>%
  rownames_to_column(., "Code_element")
knitr::kable(Wilcox_result_comm_elements_sign, format = "html", full_width = F,col.names = c("Code_element", "W", "p-value","p-value adjusted"), digits = 10) %>%
  kable_styling(latex_options="scale_down")

summary_table.W_elements <- table.W_elements %>%
  group_by(region) %>%
  summarise(
    mean_B0103 = mean(B0103),
    sd_B0103 = sd(B0103),
    mean_B1028 = mean(B1028),
    sd_B1028 = sd(B1028),
    mean_D0302 = mean(D0302),
    sd_D0302 = sd(D0302),
    mean_D0309 = mean(D0309),
    sd_D0309 = sd(D0309),
    mean_D0501 = mean(D0501),
    sd_D0501 = sd(D0501),
    mean_D0701 = mean(D0701),
    sd_D0701= sd(D0701),
    mean_D0705 = mean(D0705),
    sd_D0705 = sd(D0705),
    mean_D0807 = mean(D0807),
    sd_D0807 = sd(D0807),
    mean_D0905 = mean(D0905),
    sd_D0905= sd(D0905),
    mean_D0911 = mean(D0911),
    sd_D0911 = sd(D0911),
    mean_S0104 = mean(S0104),
    sd_S0104= sd(S0104),
    mean_S0201 = mean(S0201),
    sd_S0201 = sd(S0201),
  ) %>%
  filter(region %in% regions)
```
```{r}
# B0103 - Nucleic acid biosynthesis - UDP/UTP
ce1<- ggplot(GIFTs_elements_community_df, aes(x = region, y = B0103, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "B0103 - Nucleic acid biosynthesis - UDP/UTP",
    x = "Region",
    y = "B0103"
  )
# B1028 - Antibiotic biosynthesis - Pyocyanin
ce2<- ggplot(GIFTs_elements_community_df, aes(x = region, y = B1028, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "B1028 - Antibiotic biosynthesis - Pyocyanin",
    x = "Region",
    y = "B1028"
  )
# D0302 - Sugar degradation - Sucrose
ce3<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0302, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0302 - Sugar degradation - Sucrose",
    x = "Region",
    y = "D0302"
  )
# D0309 - Sugar degradation - Galactose
ce4<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0309, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0309 - Sugar degradation - Galactose",
    x = "Region",
    y = "D0309"
  )
# D0501 - Amino acid degradation - Serine
ce5<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0501, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0501 - Amino acid degradation - Serine",
    x = "Region",
    y = "D0501"
  )
# D0701 - Alcohol degradation - 2,3-Butanediol
ce6<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0701, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0701 - Alcohol degradation - 2,3-Butanediol",
    x = "Region",
    y = "D0701"
  )
# D0705 - Alcohol degradation - Propylene glycol
ce7<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0705, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0705 - Alcohol degradation - Propylene glycol",
    x = "Region",
    y = "D0705"
  )
# D0807 - Xenobiotic degradation - Catechol
ce8<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0807, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0807 - Xenobiotic degradation - Catechol",
    x = "Region",
    y = "D0807"
  )
# D0905 - Antibiotic degradation - Streptogramin
ce9<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0905, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0905 - Antibiotic degradation - Streptogramin",
    x = "Region",
    y = "D0905"
  )
# D0911 - Antibiotic degradation - Lincosamide
ce10<- ggplot(GIFTs_elements_community_df, aes(x = region, y = D0911, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "D0911 - Antibiotic degradation - Lincosamide",
    x = "Region",
    y = "D0911"
  )
# S0104 - Cellular structure - Lipoteichoic acid
ce11<- ggplot(GIFTs_elements_community_df, aes(x = region, y = S0104, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "S0104 - Cellular structure - Lipoteichoic acid",
    x = "Region",
    y = "S0104"
  )
# S0201 - Appendages - Flagellum
ce12<- ggplot(GIFTs_elements_community_df, aes(x = region, y = S0201, group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "S0201 - Appendages - Flagellum",
    x = "Region",
    y = "S0201"
  )
# Joint figure
library(gridExtra)
grid.arrange(arrangeGrob(ce1, ce2,ce3, ce4, ce5, ce6, ce7, ce8, ce9, ce10, ce11, ce12, ncol = 4))
```
# UPSET RDA
```{r}
## Upset visualization: MAGs in different locations and shared among locations
library(UpSetR)
locationcolors=c('#c4d7d1','#408892')

count_table_rda_binary <- ifelse(count_table_cov_size > 0, 1, 0) %>%
  as.data.frame()
count_table_rda_binary_ittoq <- count_table_rda_binary[, colnames(count_table_rda_binary) %in% Ittoq_samples]
count_table_rda_binary_ittoq$sum_of_rows <- rowSums(count_table_rda_binary_ittoq)
count_table_rda_binary_ittoq <- rownames_to_column(count_table_rda_binary_ittoq, var="genome")
ittoq_summed <- count_table_rda_binary_ittoq[, c('sum_of_rows', "genome")]
colnames(ittoq_summed)[colnames(ittoq_summed) == 'sum_of_rows'] <- 'Ittoq'

count_table_rda_binary_daneborg <- count_table_rda_binary[, colnames(count_table_rda_binary) %in% Daneborg_samples]
count_table_rda_binary_daneborg$sum_of_rows <- rowSums(count_table_rda_binary_daneborg)
count_table_rda_binary_daneborg <- rownames_to_column(count_table_rda_binary_daneborg, var="genome")
daneborg_summed <- count_table_rda_binary_daneborg[, c('sum_of_rows', "genome")]
colnames(daneborg_summed)[colnames(daneborg_summed) == 'sum_of_rows'] <- 'Daneborg'

upsetrda_1 <- merge(ittoq_summed, daneborg_summed, by = "genome", all =T )
upsetrda_1 <- column_to_rownames(upsetrda_1, var='genome')
upset_binary <- ifelse(upsetrda_1 > 0, 1, 0)

#pdf("figures/MAG_intersection.pdf",width=8,height=6, onefile=F)
upset(as.data.frame(upset_binary),
      keep.order = T,
      sets = rev(c("Daneborg","Ittoq")),
      sets.bar.color= rev(locationcolors),
      mb.ratio = c(0.55, 0.45), order.by = "freq")
#dev.off()
```
##RDA
```{r physeq object with normalized counts, structural zeros included}
# OTU table
MAGotu_zeros <- count_table_cov_size %>% # genome size normalized counts
  column_to_rownames("genome") %>%
  otu_table(., taxa_are_rows = TRUE)
# Tax table
taxonomyclean1 <- taxonomyclean[,-1]
taxmatrix <- as.matrix(taxonomyclean1[1:8])
rownames(taxmatrix) <- taxonomyclean$genome
taxtable <- tax_table(taxmatrix) 
#Metadata
metadata <- na.omit(metadata)
rownames(metadata) <- NULL
metadata.pre <- column_to_rownames(metadata, "sample")
#Sample table
sample_tab <- sample_data(metadata.pre)
# Tree
tree <- phytools::force.ultrametric(tree, method = "extend")
treephylo = phyloseq::phy_tree(tree)
## Merge into phyloseq object
physeq_rda <- phyloseq(MAGotu_zeros, treephylo, sample_tab,taxtable)
```
```{r}
## RDA
count_table_rda <- data.frame(physeq_rda@otu_table)
count_table_rda_t <- data.frame(t(count_table_rda))
hel1 <- decostand(count_table_rda_t, "hellinger")

physeq_rda_clr <- microbiome::transform(physeq_rda, 'clr')

count_rda_clr <- data.frame(physeq_rda_clr@otu_table)
count_rda_crl_t <- data.frame(t(count_rda_clr))
metadata_rda_crl <- data.frame(physeq_rda_clr@sam_data)
metadata_rda_crl <- rownames_to_column(metadata_rda_crl, "sample")
metadata_rda_crl <- metadata_rda_crl[match(rownames(count_rda_crl_t),metadata_rda_crl$sample),]
mean(rownames(count_rda_crl_t)==metadata_rda_crl$sample)
metadata_rda_crl$region=factor(metadata_rda_crl$region)
design <- metadata_rda_crl[, c("region", "sex")]

rda.model <- rda(count_rda_crl_t ~ region, data = design)# Run the model
rda.model.hel <- rda(hel1 ~ region, data = design)# Run the model
#summary(rda.model)
# Anova
anova(rda.model)
anova(rda.model.hel)
# anova(hel_rda,by="axis")
# summary(hel_rda)
RsquareAdj(rda.model)
RsquareAdj(rda.model.hel)
anova.cca(rda.model, step = 1000)
anova.cca(rda.model, step = 1000, by = "term")
anova.cca(rda.model, step = 1000, by = "axis")
perc <- round(100*(summary(rda.model)$cont$importance[2, 1:2]), 2)
## extract scores - these are coordinates in the RDA space
sc_si <- vegan::scores(rda.model, display="sites", choices=c(1,2), scaling=1)
sc_sp <- vegan::scores(rda.model, display="species", choices=c(1,2), scaling=1)
sc_bp <- vegan::scores(rda.model, display="bp", choices=c(1, 2), scaling=1)

# Prepare data for plotting
rda_sc_wa <- vegan::scores(rda.model,
                           display = "wa",
                           scaling = 1
)
rda_sc_cn <- data.frame(vegan::scores(rda.model,
                                      display = "cn",
                                      scaling = 1
))
rda_sc_sp <- data.frame(vegan::scores(rda.model,
                                      display = "sp",
                                      scaling = 2
))

# Select MAGs from quantile
ASV_Scores_RDA1quantiles <- quantile(rda_sc_sp$RDA1,
                                     probs = c(0.01, 0.99)
)
HighFit_ASVs <- rda_sc_sp[
  rda_sc_sp$RDA1 < ASV_Scores_RDA1quantiles[1] |
    rda_sc_sp$RDA1 > ASV_Scores_RDA1quantiles[2],
]

HighFit_ASVNames <- rownames(HighFit_ASVs)
species_index <- data.frame(
  Species = HighFit_ASVNames,
  Index = 1:length(HighFit_ASVNames)
)

# combine dataframes
all_dataframe <- data.frame(rda_sc_wa, design)

# Plot
ggplot(all_dataframe, aes(y = PC1, x = RDA1)) +
  geom_point(aes(y = PC1, x = RDA1, colour = region)) +
  xlab(paste0("RDA1 (", perc[1], "%)")) +
  ylab(paste0("PC1 (", perc[2], "%)")) +
  geom_segment(data = HighFit_ASVs,
               aes(x = 0, xend = RDA1, y = 0, yend = PC1),
               arrow = arrow(length = unit(0.25, "cm")),
               color = "grey") +
  geom_point(data = HighFit_ASVs,
             aes(y = PC1, x = RDA1),
             show.legend = FALSE,
             color = "grey") +
  geom_label_repel(data = HighFit_ASVs,
                   aes(y = PC1, x = RDA1,label = species_index$Species),
                   size = 3,
                   show.legend = FALSE,
                   colour = "grey",
                   max.overlaps = 20) +
  scale_shape_discrete(solid = TRUE) +
  theme_bw() +
  theme(
    plot.title = element_text(size = 10),
    legend.position = "none"
  )

```
#RQL
```{r}
# Abundance table
genome_counts=count_table_cov_size
genome_counts <- genome_counts %>%
  column_to_rownames(., "genome")
# Metadata
metadata=all_metadata
# MAG stats (mag_length)
counts_raw=mags_table

stats <- counts_raw[,c(1,11)] %>%
  as.data.frame()%>%
  column_to_rownames(., "genome")

# MAG taxonomy
mags_table <- taxonomyclean
```
## Standardization and correction
```{r}
mag_weighted <- round(
  sweep(genome_counts, MARGIN = 1, 1000, `*`), 0 
)

mag_weighted <- mag_weighted %>%
  rownames_to_column(., var = "mag") %>%
  pivot_longer(!mag, names_to = "sample", values_to = "counts") %>%
  pivot_wider(names_from = "mag", values_from = "counts") %>%
  arrange(sample) %>%
  column_to_rownames(var = "sample")

# metadata summary
design <- metadata[, c(
  "sample", "region", "sex"
)]
design$region<-0
design$region[design$region=="Daneborg"]<-1
for(i in 1:ncol(design)){
  if(is.character(design[,i])){
    design[,i]<-factor(design[,i])
  }
}
design$sex<-0
design$sex[design$sex=="Male"]<-1
for(i in 1:ncol(design)){
  if(is.character(design[,i])){
    design[,i]<-factor(design[,i])
  }
}
# MAG functional annotations (DAMMA)
functions <- GIFTs_functions %>%
  as.data.frame()
functions <- functions[,colSums(functions)>0]
```
## RLQ analysis (functions)
```{r}
mag_weighted_rel <- decostand(mag_weighted,method = "total")
mag_weighted_rel_log <- log1p(mag_weighted_rel)
```
```{r}
#Correspondence analysis (dudi.coa) is applied to the species table.
AnalysisL_functions <- dudi.coa(mag_weighted_rel_log, scannf = FALSE)

#Separate analyses of traits and environmental variables should be weighted by the sites and species weights derived from the previous correspondence analysis.
#Principal Component Analysis (PCA) if all variables are continuous
# AnalysisQ_functions <- dudi.pca(functions, row.w = AnalysisL_functions$cw,scannf = FALSE)
#Hill and smith analysis if there is a mix of continuous and categorical variables. Hill and smith analysis is calculated with dudi.hillsmith()
AnalysisR_functions <- dudi.hillsmith(design[,c(5,6)], row.w = AnalysisL_functions$lw,scannf = FALSE)#function that allows to consider mix of different types of variables
```
```{r}
rlq.functions <- rlq(AnalysisR_functions, AnalysisL_functions, AnalysisQ_functions,scannf = FALSE)
summary(rlq.functions)
```
#Kraken
```{r Percent Eukaryotic summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Eukaryota_.)
# Range 3-13 % reads mapped to Eukaryota
# Med = 8.5%, Mean = 8.517

# Shapiro-Wilk test for normality
shapiro_results_Euk <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Eukaryota_.)$p.value) # Results: p = 0.00615 Daneborg, p = 0.0369 Ittoq

# Wilcoxon Exact rank test
library(exactRankTests)
wilcox_exact_results_Euk <- exactRankTests::wilcox.exact(Eukaryota_. ~ region, data = all_metadata)
print(wilcox_exact_results_Euk) # Results: p = 0.6329 # THEREFORE no sig difference in % Eukaryotic reads between regions

# Plot
ggplot(all_metadata, aes(x = region, y = Eukaryota_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Eukaryota",
    x = "Region",
    y = "% of unmapped reads"
  )
```
**For all remaining summaries, values are reported as the % of Eukaryotic data mapped to each taxa, and "no" means that PR found no parasites, whereas "yes" indicates that samples tested positive**
Cryptosporidium
```{r Cryptosporidium summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Cryptosporidium_.)
# Range 0.00300 - 0.02000 , note no samples with 0s
# Med = 0.00600, Mean = 0.00669

# Shapiro-Wilk test for normality (grouped by region)
shapiro_results_Crypto_R <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Cryptosporidium_.)$p.value) # Results: p = 0.0631 Daneborg, p = 0.0000802 Ittoq
shapiro_results_Crypto_PR <- all_metadata %>%
  filter(PR_cryptosporidium != "excluded") %>%
  group_by(PR_cryptosporidium) %>%
  summarise(shapiro_p_value = shapiro.test(Cryptosporidium_.)$p.value) # Results: p =  0.00000147 for "no" , p = 0.621 for "yes"

# Wilcoxon Exact rank test
library(exactRankTests)
wilcox_exact_results_Crypto_R <- exactRankTests::wilcox.exact(Cryptosporidium_. ~ region, data = all_metadata)
print(wilcox_exact_results_Crypto_R) # Results: p = 0.007237 # THEREFORE there IS a significant difference in Crypto between regions
wilcox_results_Crypto_PR <- all_metadata %>%
  filter(PR_cryptosporidium != "excluded") %>%
  group_by(PR_cryptosporidium) %>%
  summarise(wilcox_p_value = wilcox.test(Cryptosporidium_. ~ PR_cryptosporidium, data = .)$p.value) # Results: p =  0.483 # THEREFORE no significant difference detected in Cryptosporidium values, when grouped by PR results 

# Plot by region
ggplot(all_metadata, aes(x = region, y = Cryptosporidium_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Cryptosporidium",
    x = "Region",
    y = "% of Eukaryotic reads"
  )

# Plot by PR results
ggplot(all_metadata %>% filter(PR_cryptosporidium != "excluded"),
       aes(x = PR_cryptosporidium, y = Cryptosporidium_., group = PR_cryptosporidium, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Cryptosporidium",
    x = "PR Results",
    y = "% of Eukaryotic reads"
  )

############ what happens when we remove the outlier at 0.02?
```
Giardia
```{r Giardia summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Giardia_.)
# Range 0 - 0.004
summary((subset(all_metadata, Giardia_. != 0)$Giardia_.))
# range 0.00001 - 0.004 ; when negative samples are removed
means_Giardia_region <- all_metadata %>%
  group_by(region) %>%
  summarize(mean_Giardia = mean(Giardia_., na.rm = TRUE)) # means = 0.00035 Daneborg, 0.00047 Ittoq

# Shapiro-Wilk test for normality (grouped by region)
shapiro_results_Giardia_R <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Giardia_.)$p.value) # Results: p = 6.32e-10 Daneborg, p = 4.12e- 5 Ittoq
shapiro_results_Giardia_PR <- all_metadata %>%
  filter(PR_giardia != "excluded") %>%
  group_by(PR_giardia) %>%
  summarise(shapiro_p_value = shapiro.test(Giardia_.)$p.value) # Results: p =  1.52e-11 for "no" , p = 6.72e- 1 for "yes"

# Wilcoxon Exact rank test
library(exactRankTests)
wilcox_exact_results_Giardia_R <- all_metadata %>%
  filter(Giardia_. != 0) %>%
  wilcox.exact(Giardia_. ~ region, data = .) # Results: p = p-value = 0.01454 significant difference in Giardia between regions

wilcox_results_Giardia_PR <- all_metadata %>%
  filter(PR_giardia != "excluded") %>%
  filter(Giardia_. != 0) %>%
  summarise(wilcox_p_value = wilcox.test(Giardia_. ~ PR_giardia, data = .)$p.value)
# Results: p =  0.0521357 no significant difference detected in Giardia values, when grouped by PR results 

# Plot by region
ggplot(all_metadata %>% filter(Giardia_. != 0), aes(x = region, y = Giardia_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Giardia",
    x = "Region",
    y = "% of Eukaryotic reads"
  )

# Plot by PR results
ggplot(all_metadata %>% filter(PR_giardia != "excluded") %>% filter(Giardia_. != 0),
       aes(x = PR_giardia, y = Giardia_., group = PR_giardia, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Giardia",
    x = "Classical Parasitology",
    y = "% of Eukaryotic reads"
  )
```
Sarcocystis
```{r Sarcocystis summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Sarcocystis_.)
# Range 0 - 3.000e-04 
# Med = 0, Mean = 6.207e-06

# only 3 samples (2 Ittoq, 1 Daneborg) had Sarcocystis DNA. 
```
Toxoplasma_gondii
```{r Toxoplasma_gondii summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Toxoplasma_gondii_.)
# Range 0.001 - 0.03 ; Med = 0.004, Mean = 0.004897 
means_T_gondii_region <- all_metadata %>%
  group_by(region) %>%
  summarize(mean_T_gondii = mean(Toxoplasma_gondii_., na.rm = TRUE)) # Daneborg 0.00445 ; Ittoq 0.00534

# Shapiro-Wilk test for normality (grouped by region)
shapiro_results_Toxoplasma_gondii_R <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Toxoplasma_gondii_.)$p.value) # Results: p = 0.00431 Daneborg, p = 0.00000000427 Ittoq # THEREFORE cannot assume normality for whole dataset

# Wilcoxon Exact rank test
wilcox_exact_results_Toxoplasma_gondii_R <- exactRankTests::wilcox.exact(Toxoplasma_gondii_. ~ region, data = all_metadata)
print(wilcox_exact_results_Toxoplasma_gondii_R)
# Results: p-value = 0.6477 NO significant difference in Toxoplasma_gondii between regions

# serological testing for only 5 dogs by P/R, only one positive, so too small to compare

# Plot by region
ggplot(all_metadata, aes(x = region, y = Toxoplasma_gondii_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Toxoplasma_gondii",
    x = "Region",
    y = "% of Eukaryotic reads"
  )
```
No toxacaris detected with kraken

Trichinella
```{r Trichinella summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Trichinella_.)
# Range 0.005 - 0.3 Med = 0.01, Mean = 0.01429
means_Trichinella_region <- all_metadata %>%
  group_by(region) %>%
  summarize(mean_Trichinella = mean(Trichinella_., na.rm = TRUE)) # Daneborg 0.0203 ; Ittoq 0.00831

# Shapiro-Wilk test for normality (grouped by region)
shapiro_results_Trichinella_R <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Trichinella_.)$p.value) # Results: p = 2.12e-11 Daneborg, p = 9.37e- 4 Ittoq

# serological testing for only 5 dogs by P/R, all positive, so no way to compare

# Wilcoxon Exact rank test
wilcox_exact_results_Trichinella_R <- exactRankTests::wilcox.exact(Trichinella_. ~ region, data = all_metadata)
print(wilcox_exact_results_Trichinella_R) # Results: p = 0.0002889 significant difference in Trichinella between regions

# Plot by region
ggplot(all_metadata, aes(x = region, y = Trichinella_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Trichinella",
    x = "Region",
    y = "% of Eukaryotic reads"
  ) +
    coord_cartesian(ylim = c(0, 0.03))
```
Taenia
```{r Taenia summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Taenia_.)
# Range 0 - 0.008
# Med = 0, Mean = 0.0004343
mean(subset(all_metadata, Taenia_. != 0)$Taenia_.) #0.001145
means_Taenia_region <- all_metadata %>% 
  filter(Taenia_. != 0) %>%
  group_by(region) %>%
  summarize(mean_Taenia = mean(Taenia_., na.rm = TRUE)) # Daneborg 0.00101 ; Ittoq 0.0017

# Shapiro-Wilk test for normality (grouped by region)
shapiro_results_Taenia_R <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Taenia_.)$p.value) #Results: p = 1.03e- 9 Daneborg, p = 1.45e-11 Ittoq

# Wilcoxon Exact rank test
wilcox_exact_results_Taenia_R <- all_metadata %>%
  filter(Taenia_. != 0) %>% # hash this line to calculate WITH negative dogs included
  wilcox.exact(Taenia_. ~ region, data = .) # p-value = 0.4643, no significant difference in Taenia between regions (with zeros removed) ; p-value = 7.409e-06, sig difference when zeros included

# PR results only found 1 dog with Taenia, therefore unable to group by their results. 

# Plot by region
ggplot(all_metadata %>% filter(Taenia_. != 0), aes(x = region, y = Taenia_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Taenia",
    x = "Region",
    y = "% of Eukaryotic reads"
  )
```
Uncinaria
```{r Percent Uncinaria summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Uncinaria_.)
# Range 0 - 0.02 
# Med = 0, Mean = 0.0003843
summary((subset(all_metadata, Uncinaria_. != 0)$Uncinaria_.))
means_Uncinaria_region <- all_metadata %>% 
  filter(Uncinaria_. != 0) %>%
  group_by(region) %>%
  summarize(mean_Uncinaria = mean(Uncinaria_., na.rm = TRUE)) # Daneborg 0.0012 ; Ittoq 0.000072

# Shapiro-Wilk test for normality
shapiro_results_Uncinaria <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Uncinaria_.)$p.value) # Results: p = 1.71e-11 Daneborg, p =  3.81e-10 Ittoq

# Wilcoxon Exact rank test
wilcox_exact_results_Uncinaria_R <- all_metadata %>%
  filter(Uncinaria_. != 0) %>% # hash this line to calculate WITH negative dogs included
  wilcox.exact(Uncinaria_. ~ region, data = .) # p = 0.1303, NO significance when zeros are excluded ; p =0.0001526 # THEREFORE YES sig difference between regions (with zeros included)

# Plot
ggplot(all_metadata, aes(x = region, y = Uncinaria_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Uncinaria",
    x = "Region",
    y = "% of Euk reads"
  ) +
      coord_cartesian(ylim = c(0, 0.00025))
```
Echinococcus
```{r Percent Echinococcus summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Echinococcus_.) # Range 0.003 - 0.08  Med = 0.00850, Mean = 0.01167
means_Echinococcus_region <- all_metadata %>% 
  filter(Echinococcus_. != 0) %>%
  group_by(region) %>%
  summarize(mean_Echinococcus = mean(Echinococcus_., na.rm = TRUE)) # Daneborg 0.013 ; Ittoq 0.010

# Shapiro-Wilk test for normality
shapiro_results_Echinococcus <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Echinococcus_.)$p.value) # Results: p = 0.00000000897 Daneborg, p =  0.00000784 Ittoq

# Wilcoxon Exact rank test
wilcox_exact_results_Echinococcus <- exactRankTests::wilcox.exact(Echinococcus_. ~ region, data = all_metadata)
print(wilcox_exact_results_Echinococcus) # Results: p =0.312 THEREFORE no sig difference between regions

# Plot
ggplot(all_metadata, aes(x = region, y = Echinococcus_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Echinococcus",
    x = "Region",
    y = "% of Euk reads"
  )
```
Cestoda
```{r Percent Cestoda summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Cestoda_.) # Range 0.06 - 0.9  Med = 0.1, Mean = 0.1129

# Shapiro-Wilk test for normality
shapiro_results_Cestoda <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Cestoda_.)$p.value) # Results: p = 3.39e-11 Daneborg, p =  1.03e- 9 Ittoq

# Wilcoxon Exact rank test
wilcox_exact_results_Cestoda <- exactRankTests::wilcox.exact(Cestoda_. ~ region, data = all_metadata)
print(wilcox_exact_results_Cestoda) # Results: p =0.04305  sig difference between regions
```
Nematoda
```{r Percent Nematoda summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
summary(all_metadata$Nematoda_.) # Range   0.02 - 0.4  Med = 0.3, Mean = 0.259

# Shapiro-Wilk test for normality
shapiro_results_Nematoda <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Nematoda_.)$p.value) # Results: p = 0.00000112 Daneborg, p =   0.00000262 Ittoq

# Wilcoxon Exact rank test
wilcox_exact_results_Nematoda <- exactRankTests::wilcox.exact(Nematoda_. ~ region, data = all_metadata)
print(wilcox_exact_results_Nematoda) # Results: p =0.3481 no sig difference between regions
```
# Viruses
Viral overall
```{r Virus}
summary(all_metadata$Virus_.) # Range 0.3-4.0 % reads mapped to Virus Med = 0.07%, Mean = 0.225

# Shapiro-Wilk test for normality
shapiro_results_Virus <- all_metadata %>%
  group_by(region) %>%
  summarise(shapiro_p_value = shapiro.test(Virus_.)$p.value) # Results: p = 0.000237 Daneborg, p = 0.00000000206 Ittoq

# Wilcoxon Exact rank test
library(exactRankTests)
wilcox_exact_results_Virus <- exactRankTests::wilcox.exact(Virus_. ~ region, data = all_metadata)
print(wilcox_exact_results_Virus) # Results: p = 0.002321 YES sig diff in % Virus reads between regions

# Plot
ggplot(all_metadata, aes(x = region, y = Virus_., group = region, color = region)) +
  geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank()
  ) +
  labs(
    title = "Virus",
    x = "Region",
    y = "% of unmapped reads"
  ) + coord_cartesian(ylim = c(0, 1) )
```
```{r Parvo Adeno Pap}
# Parvo
library(exactRankTests)
summary(all_metadata$PARVO_Parvoviridae_.)
wilcox_exact_results_Parvo <- exactRankTests::wilcox.exact(all_metadata$PARVO_Parvoviridae_. ~ region, data = all_metadata)
print(wilcox_exact_results_Parvo) # Results: p = 7.547e-09 sig diff in % Viral reads mapped to Parvo between regions
means_Parvoviridae_region <- all_metadata %>% 
  filter(PARVO_Parvoviridae_. != 0) %>%
  group_by(region) %>%
  summarize(mean_PARVO_Parvoviridae_. = mean(PARVO_Parvoviridae_., na.rm = TRUE)) # Daneborg 0.42 ; Ittoq 0.41 (without zeros) / Daneborg 0.41 ; Ittoq 0.38 (with zeros)

# Adenovirus
library(exactRankTests)
wilcox_exact_results_Adeno <- exactRankTests::wilcox.exact(all_metadata$ADENO_Adenoviridae_. ~ region, data = all_metadata)
print(wilcox_exact_results_Adeno) # Results: p = 0.0001022  sig diff in % Viral reads mapped to Adeno between regions
means_Adeno_region <- all_metadata %>% 
  filter(ADENO_Adenoviridae_. != 0) %>% # hash out to calculate with zeros
  group_by(region) %>%
  summarize(mean_ADENO_Adenoviridae_. = mean(ADENO_Adenoviridae_., na.rm = TRUE)) # Daneborg 0.012 ; Ittoq 0.082 (without zeros) / Daneborg 0.0081 , 0.076 (with zeros)

#Papilloma
library(exactRankTests)
wilcox_exact_results_Pap <- exactRankTests::wilcox.exact(all_metadata$PAP_Papillomaviridae_. ~ region, data = all_metadata)
print(wilcox_exact_results_Pap) # Results: p = 8.691e-05 sig diff in % Viral reads mapped to Pap between regions
means_Adeno_region <- all_metadata %>% 
  filter(PAP_Papillomaviridae_. != 0) %>% # hash out to calculate with zeros
  group_by(region) %>%
  summarize(mean_PAP_Papillomaviridae_. = mean(PAP_Papillomaviridae_., na.rm = TRUE)) # Daneborg 0.02 , Ittoq 0.21 (without zeros) / values do not change with zeros added
```
## Fishers Exact testing:
```{r inferred species}
inferred_species <- read.csv("/Users/elsa/Desktop/THESIS/Thesis/inferred_species.csv") %>% as.data.frame()
columns_to_select <- setdiff(names(inferred_species), "region")

# Perform left join excluding the 'region' column
all_metadata <- dplyr::left_join(all_metadata, inferred_species %>% dplyr::select(all_of(columns_to_select)), by = "sample")
```
```{r counts of higher resolution taxonomic info}
# Parvo
count_Parvo <- aggregate(PARVO_Parvoviridae ~ region, data = all_metadata, FUN = sum) # Daneborg = 28, Ittoq = 27
count_Bocaparvovirus <- aggregate(PARVO_Bocaparvovirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 23, Ittoq = 27
count_Human_Bocavirus <- aggregate(PARVO_Human_Bocavirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 19, Ittoq = 1
count_Bufavirus_1 <- aggregate(PARVO_Bufavirus_1 ~ region, data = all_metadata, FUN = sum) # Daneborg 9, Ittoq = 10
count_Canine_Minute <- aggregate(PARVO_Canine_Minute ~ region, data = all_metadata, FUN = sum) # Daneborg = 0, Ittoq = 1
count_Canine_bufavirus <- aggregate(PARVO_Canine_bufavirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 0, Ittoq = 1
count_Canine_protoparvovirus <- aggregate(PARVO_Canine_protoparvovirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 0, Ittoq = 1
count_Canine_parvovirus <- aggregate(PARVO_Canine_parvovirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 1, Ittoq = 0

# Adeno
count_Adeno <- aggregate(ADENO_Adenoviridae ~ region, data = all_metadata, FUN = sum) # Daneborg = 19 Ittoq = 27
count_Canine_mastadenovirus <- aggregate(ADENO_Canine_mastadenovirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 0, Ittoq = 3
count_Human_adenovirus <- aggregate(ADENO_Human_adenovirus ~ region, data = all_metadata, FUN = sum) # Daneborg = 3, Ittoq = 24

# Pap
count_Pap <- aggregate(PAP_Papillomaviridae ~ region, data = all_metadata, FUN = sum) #Daneborg = 28 Ittoq = 29
count_HPV <- aggregate(PAP_HPV ~ region, data = all_metadata, FUN = sum) # Daneborg = 28, Ittoq = 29
count_Tick_assoc_papillomavirus_lsx <- aggregate(PAP_Tick_assoc_papillomavirus_lsx ~ region, data = all_metadata, FUN = sum) # Daneborg 0, Ittoq = 25
```
```{r contingency tables}
disease_columns <- c("S_capracanis", "S_tuagulusi", "S_hirsuta", "G_intestinalis", "G_peramelis", "G_muris", "Tr_spiralis", "Tr_OTHER", "Tr_pathgoniensis", "Tr_pseudospiralis", "Ta_laticollis", "Ta_asiatica", "Ta_solium", "Ta_OTHER", "Ta_krabbei", "Ta_arctos", "Ta_multiceps", "Ta_saginata", "Ta_madoquae", "E_granulosus", "E_multilocularis", "C_ubiquitum", "C_parvum", "C_hominis_TU502", "C_proventriculi", "C_muris_RN66", "C_OTHER", "C_wrairi", "C_canis", "PARVO_Parvoviridae", "PARVO_Bocaparvovirus", "PARVO_Human_Bocavirus", "PARVO_Bufavirus_1" , "PARVO_Canine_Minute", "PARVO_Canine_bufavirus", "PARVO_Canine_bufavirus", "PARVO_Canine_protoparvovirus", "PARVO_Canine_parvovirus", "ADENO_Adenoviridae", "ADENO_Human_adenovirus", "ADENO_Canine_mastadenovirus", "PAP_HPV", "PAP_Papillomaviridae", "PAP_Tick_assoc_papillomavirus_lsx")

contingency_table <- table(all_metadata$region, all_metadata$C_meleagridis) # change out disease to test all

# Run Fisher's exact test
fisher_result <- fisher.test(contingency_table)

# Print the results
print(fisher_result)
```